%% Generated by Sphinx.
\def\sphinxdocclass{article}
\documentclass[letterpaper,10pt,english]{sphinxhowto}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{times}
\expandafter\ifx\csname T@LGR\endcsname\relax
\else
% LGR was declared as font encoding
  \substitutefont{LGR}{\rmdefault}{cmr}
  \substitutefont{LGR}{\sfdefault}{cmss}
  \substitutefont{LGR}{\ttdefault}{cmtt}
\fi
\expandafter\ifx\csname T@X2\endcsname\relax
  \expandafter\ifx\csname T@T2A\endcsname\relax
  \else
  % T2A was declared as font encoding
    \substitutefont{T2A}{\rmdefault}{cmr}
    \substitutefont{T2A}{\sfdefault}{cmss}
    \substitutefont{T2A}{\ttdefault}{cmtt}
  \fi
\else
% X2 was declared as font encoding
  \substitutefont{X2}{\rmdefault}{cmr}
  \substitutefont{X2}{\sfdefault}{cmss}
  \substitutefont{X2}{\ttdefault}{cmtt}
\fi


\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}
\setcounter{tocdepth}{4}
\setcounter{secnumdepth}{4}


\title{copy\sphinxhyphen{}number\sphinxhyphen{}analysis}
\date{Mar 24, 2021}
\release{1.0}
\author{Anibal Morales, IAEA PBGL \sphinxhyphen{} UN FAO}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Version 1.0}
\makeindex
\begin{document}

\pagestyle{empty}

        \pagenumbering{Roman} %%% to avoid page 1 conflict with actual page
        \begin{titlepage}
            \vspace*{10mm} %%% * is used to give space from top
            \flushright\textbf{\Huge {PBGL CNV-seq Analysis v1.0\\}}
            \vspace{0mm} %%% * is used to give space from top
            \textbf{\Large {A Laboratory Manual\\}}
            \vspace{50mm}
            \textbf{\Large {Anibal E. Morales-Zambrana\\}}
            \vspace{10mm}
            \textbf{\Large {Plant Breeding and Genetics Laboratory\\}}
            \vspace{0mm}
            \textbf{\Large {FAO/IAEA Joint Division\\}}
            \vspace{0mm}
            \textbf{\Large {Seibersdorf, Austria\\}}
	    \vspace{10mm}
            \normalsize Created: March, 2021\\
            \vspace*{0mm}
            \normalsize  Last updated: 24 March 2021
            %% \vfill adds at the bottom
            \vfill
            \small\flushleft {{\textbf {Please note:}} \textit {This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views
expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.}}
        \end{titlepage}
        \pagenumbering{arabic}
        \newcommand{\sectionbreak}{\clearpage}

\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\section{Background}
\label{\detokenize{index:background}}
\sphinxstylestrong{{[}DRAFT{]}}

Copy number variation (CNV) analysis using CNV\sphinxhyphen{}seq, R, Jupyter Notebooks, Miniconda3, and Git.

\begin{sphinxadmonition}{note}{Note:}
This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.
\end{sphinxadmonition}


\section{Installations}
\label{\detokenize{index:installations}}
Before installing any necessary software, it is recommended to check if the computer is running 32\sphinxhyphen{}bit or 64\sphinxhyphen{}bit. Run the following to verify the system:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} uname \PYGZhy{}m
\end{sphinxVerbatim}


\subsection{Miniconda3 (conda)}
\label{\detokenize{index:miniconda3-conda}}
Download the Miniconda3, or simply “conda”, installer:
\begin{itemize}
\item {} 
\sphinxhref{https://docs.conda.io/en/latest/miniconda.html\#linux-installers}{Miniconda3 installer for Linux}

\end{itemize}

Run the downloaded installer:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} bash Miniconda3\PYGZhy{}latest\PYGZhy{}Linux\PYGZhy{}x86\PYGZus{}64.sh
\end{sphinxVerbatim}

Open a new terminal window for conda to take effect. Verify the installation in new terminal window with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda list
\end{sphinxVerbatim}


\subsection{Git with conda}
\label{\detokenize{index:git-with-conda}}
Git will be installed first to clone locally (download a copy to your local computer) the CNV\sphinxhyphen{}seq repository from GitHub. To do so, run the following:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda install \PYGZhy{}c anaconda git
\end{sphinxVerbatim}

After the installation, clone the CNV\sphinxhyphen{}seq repository to the local computer in the desired directory.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git clone https://github.com/amora197/copy\PYGZhy{}number\PYGZhy{}analysis.git
\end{sphinxVerbatim}

Verify that the installation is complete by listing the files in the directory.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ls \PYGZhy{}l
\end{sphinxVerbatim}

A folder called \sphinxstylestrong{copy\sphinxhyphen{}number\sphinxhyphen{}analysis} should be listed in the directory.


\subsection{Required Libraries with conda}
\label{\detokenize{index:required-libraries-with-conda}}
CNV\sphinxhyphen{}seq has multiple dependencies, listed below:
\begin{itemize}
\item {} 
Git

\item {} 
R
\begin{itemize}
\item {} 
configr

\item {} 
ggplot2

\item {} 
BiocManager

\item {} 
Bioconductor\sphinxhyphen{}GenomicAlignments

\end{itemize}

\item {} 
Jupyter Notebook
\begin{itemize}
\item {} 
IRkernel

\end{itemize}

\end{itemize}

There are two ways to install the rest of the necessary libraries to run CNV\sphinxhyphen{}seq: automatically or manually. The former is slower, providing a long coffee break while the conda installations run. The latter proves a faster way to get the tool up\sphinxhyphen{}and\sphinxhyphen{}running.


\subsubsection{Automatically (slower)}
\label{\detokenize{index:automatically-slower}}
Two YAML files, \sphinxstylestrong{environment.yml} and \sphinxstylestrong{libraries.yml}, are provided to automatically create a conda environment and install the dependent libraries. The former creates the conda environment, along R, Jupyter Notebook, and the R\sphinxhyphen{}kernel in Jupyter. The latter installs dependent R libraries. Run \sphinxstylestrong{environment.yml}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda env create \PYGZhy{}\PYGZhy{}file envs/environment.yml
\end{sphinxVerbatim}

Once done, the created environment can be verified running:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda env list
\end{sphinxVerbatim}

Activate the created environment (\sphinxstylestrong{cnv\sphinxhyphen{}seq}) and run \sphinxstylestrong{libraries.yml}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda activate cnv\PYGZhy{}seq
\PYGZdl{} conda env update \PYGZhy{}\PYGZhy{}file envs/libraries.yml
\end{sphinxVerbatim}

Once done, all the necessary packages should be installed. This can be verified with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda list
\end{sphinxVerbatim}


\subsubsection{Manually (faster)}
\label{\detokenize{index:manually-faster}}
To manually create and activate an environment, run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda create \PYGZhy{}\PYGZhy{}name cnv\PYGZhy{}seq
\end{sphinxVerbatim}

Activate the virtual environment with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda activate cnv\PYGZhy{}seq
\end{sphinxVerbatim}

Start running the installations of the necessary libraries, paying attention to the prompts for each one:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda install \PYGZhy{}c conda\PYGZhy{}forge r\PYGZhy{}base=4.0
\PYGZdl{} conda install \PYGZhy{}c anaconda jupyter
\PYGZdl{} conda install \PYGZhy{}c r r\PYGZhy{}irkernel
\PYGZdl{} conda install \PYGZhy{}y \PYGZhy{}c conda\PYGZhy{}forge r\PYGZhy{}biocmanager
\PYGZdl{} conda install \PYGZhy{}y \PYGZhy{}c bioconda bioconductor\PYGZhy{}genomicalignments
\PYGZdl{} conda install \PYGZhy{}y \PYGZhy{}c pcgr r\PYGZhy{}configr
\PYGZdl{} conda install \PYGZhy{}y \PYGZhy{}c r r\PYGZhy{}ggplot2
\end{sphinxVerbatim}

Once done, all the necessary packages should be installed. This can be verified with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda list
\end{sphinxVerbatim}


\section{Running a Jupyter Notebook}
\label{\detokenize{index:running-a-jupyter-notebook}}
To access the Jupyter Notebooks, run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} jupyter notebook
\end{sphinxVerbatim}

This command will start a Jupyter Notebook session inside the directory the command is run. The user can navigate between directories, visualize files, and edit files in the browser by clicking on directories or files, respectively.

Look for the directory \sphinxstylestrong{copy\sphinxhyphen{}number\sphinxhyphen{}analysis} and click on it. Click on \sphinxstylestrong{jupyter\sphinxhyphen{}notebooks} directory, which contains four directories and two Jupyter Notebooks. Here is a breakdown of each:
\begin{itemize}
\item {} 
\sphinxtitleref{config}:
\begin{itemize}
\item {} 
directory containing configuration files

\end{itemize}

\item {} 
\sphinxtitleref{helper\sphinxhyphen{}functions}:
\begin{itemize}
\item {} 
directory containing R scripts with functions to calculate and plot CNVs

\end{itemize}

\item {} 
\sphinxtitleref{images}:
\begin{itemize}
\item {} 
directory that will contain output CNV plots after running CNV\sphinxhyphen{}seq

\end{itemize}

\item {} 
\sphinxtitleref{tab\sphinxhyphen{}files}:
\begin{itemize}
\item {} 
directory that will contain two types of output tab\sphinxhyphen{}delimited files:
\begin{itemize}
\item {} 
hits used to calculate CNVs

\item {} 
CNVs per chromosome per comparison

\end{itemize}

\end{itemize}

\item {} 
two Jupyter Notebooks:
\begin{itemize}
\item {} 
RCNV\_seq\sphinxhyphen{}example.ipynb
\begin{itemize}
\item {} 
example analysis of sorghum

\end{itemize}

\item {} 
RCNV\_seq\sphinxhyphen{}template.ipynb
\begin{itemize}
\item {} 
template for the user

\end{itemize}

\end{itemize}

\end{itemize}


\section{Editing the Configuration File}
\label{\detokenize{index:editing-the-configuration-file}}
In order to run the CNV\sphinxhyphen{}seq Jupyter Notebook, the user needs to feed it with a configuration file (\sphinxstylestrong{config\sphinxhyphen{}CNVseq.yml}) that specifies the paths to the bam files, comparisons to be done, chromosomes to analyze, and parameter definitions for calculating and plotting CNVs.

The configuration file \sphinxstylestrong{config\sphinxhyphen{}CNVseq.yml} can be found in the \sphinxstylestrong{copy\sphinxhyphen{}number\sphinxhyphen{}analysis/jupyter\sphinxhyphen{}notebooks/config} directory.

\begin{sphinxadmonition}{note}{Note:}
The user needs to edit \sphinxstylestrong{config\sphinxhyphen{}CNVseq.yml} to point towards bam/bed files; specify comparisons and chromosomes to analyze; and define the parameters to calculate/plot CNVs.
\end{sphinxadmonition}

Two example configuration files are provided (\sphinxstylestrong{example1\sphinxhyphen{}config\sphinxhyphen{}CNVseq\sphinxhyphen{}coffee.yml} and \sphinxstylestrong{example2\sphinxhyphen{}config\sphinxhyphen{}CNVseq\sphinxhyphen{}sorghum.yml}). The configuration file \sphinxstylestrong{config\sphinxhyphen{}CNVseq.yml} contains multiple things to be defined by the user:
\begin{itemize}
\item {} 
\sphinxtitleref{paths}:
\begin{itemize}
\item {} 
sample names and their respective paths to \sphinxstylestrong{.bam} files

\item {} 
samples can be named as desired but the sample name must be repeated after the colons and prefixed with a \sphinxtitleref{\&} sign

\item {} 
the \sphinxtitleref{\&} prefix sign is used to reference the sample’s path in different places of the same configuration file

\item {} 
example use:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{paths}\PYG{p}{:}
  \PYG{n}{mysample}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{mysample} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{john}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{mysample}\PYG{o}{.}\PYG{n}{bam}
  \PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{123}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{123} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{john}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{123.}\PYG{n}{bam}
  \PYG{n}{potato95}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{potato95} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{john}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{potato95}\PYG{o}{.}\PYG{n}{bam}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxtitleref{bed\_path}:
\begin{itemize}
\item {} 
path to bed file if using varying window sizes

\end{itemize}

\item {} 
\sphinxtitleref{comparisons}:
\begin{itemize}
\item {} 
comparison names with respective control and mutant samples per comparison

\item {} 
each comparison can be named as desired

\item {} 
the sample names to be used as \sphinxtitleref{control} and \sphinxtitleref{mutant} need to be prefixed by a \sphinxtitleref{*} sign

\item {} 
the \sphinxtitleref{*} prefixed sign is used to extract the sample’s path defined in the \sphinxtitleref{paths} section

\item {} 
example:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{comparisons}\PYG{p}{:}
  \PYG{n}{comparison}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{:}
    \PYG{n}{control}\PYG{p}{:} \PYG{o}{*}\PYG{n}{mysample}
    \PYG{n}{mutant}\PYG{p}{:} \PYG{o}{*}\PYG{n}{potato95}
  \PYG{n}{a}\PYG{o}{\PYGZhy{}}\PYG{n}{different}\PYG{o}{\PYGZhy{}}\PYG{n}{comparison}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{278}\PYG{n}{asd}\PYG{p}{:}
    \PYG{n}{control}\PYG{p}{:} \PYG{o}{*}\PYG{n}{mysample}
    \PYG{n}{mutant}\PYG{p}{:} \PYG{o}{*}\PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{123}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxtitleref{chromosomes}:
\begin{itemize}
\item {} 
list of chromosome names to analyze separated by commas

\item {} 
otherwise all chromosomes present in bam file will be analyzed

\end{itemize}

\item {} 
\sphinxtitleref{parameters}:
\begin{itemize}
\item {} 
parameters used to create window sizes, thresholds, and plots

\item {} 
the parameter defaults are provided

\end{itemize}

\end{itemize}


\section{Running the RCNV\_seq\sphinxhyphen{}template Jupyter Notebook}
\label{\detokenize{index:running-the-rcnv-seq-template-jupyter-notebook}}
In the \sphinxstylestrong{copy\sphinxhyphen{}number\sphinxhyphen{}analysis/jupyter\sphinxhyphen{}notebooks} directory, click on \sphinxstylestrong{RCNV\_seq\sphinxhyphen{}example} and a new tab will open the notebook.

The notebook contains cells that are populated by text or code. Instructions are provided in the notebook to guide the user.

The notebook consists of 5 sections:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
User Input (MANDATORY)

\item {} 
Installing Required Libraries (optional)

\item {} 
Loading Required Libraries (MANDATORY)

\item {} 
CNV Calculations and Plotting (MANDATORY)

\item {} 
Plotting Specific Window of One Chromosome (optional)

\end{enumerate}

To run a cell, click on the corresponding cell and press \sphinxstylestrong{Ctrl + Enter} or \sphinxstylestrong{Shift + Enter}.


\section{References}
\label{\detokenize{index:references}}
\sphinxstylestrong{BMC Bioinformatics Publication}:
\begin{itemize}
\item {} 
CNV\sphinxhyphen{}seq, a new method to detect copy number variation using high\sphinxhyphen{}throughput sequencing {[}\sphinxhref{https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-80}{LINK}{]}

\end{itemize}

\sphinxstylestrong{GitHub repositories}:
\begin{itemize}
\item {} 
\sphinxhref{https://github.com/hliang/cnv-seq}{hliang/cnv\sphinxhyphen{}seq}

\item {} 
\sphinxhref{https://github.com/Bioconductor/copy-number-analysis/wiki/CNV-seq}{Bioconductor/copy\sphinxhyphen{}number\sphinxhyphen{}analysis}

\item {} 
\sphinxhref{https://github.com/amora197/copy-number-analysis}{amora197/copy\sphinxhyphen{}number\sphinxhyphen{}analysis}

\end{itemize}



\renewcommand{\indexname}{Index}
\printindex
\end{document}