%% Generated by Sphinx.
\def\sphinxdocclass{article}
\documentclass[letterpaper,10pt,english]{sphinxhowto}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{times}
\expandafter\ifx\csname T@LGR\endcsname\relax
\else
% LGR was declared as font encoding
  \substitutefont{LGR}{\rmdefault}{cmr}
  \substitutefont{LGR}{\sfdefault}{cmss}
  \substitutefont{LGR}{\ttdefault}{cmtt}
\fi
\expandafter\ifx\csname T@X2\endcsname\relax
  \expandafter\ifx\csname T@T2A\endcsname\relax
  \else
  % T2A was declared as font encoding
    \substitutefont{T2A}{\rmdefault}{cmr}
    \substitutefont{T2A}{\sfdefault}{cmss}
    \substitutefont{T2A}{\ttdefault}{cmtt}
  \fi
\else
% X2 was declared as font encoding
  \substitutefont{X2}{\rmdefault}{cmr}
  \substitutefont{X2}{\sfdefault}{cmss}
  \substitutefont{X2}{\ttdefault}{cmtt}
\fi


\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}
\setcounter{tocdepth}{4}
\setcounter{secnumdepth}{4}


\title{pbgl\sphinxhyphen{}cnvseq}
\date{Jun 21, 2021}
\release{1.0}
\author{Anibal Morales, IAEA PBGL \sphinxhyphen{} UN FAO}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Version 1.0}
\makeindex
\begin{document}

\pagestyle{empty}

        \pagenumbering{Roman} %%% to avoid page 1 conflict with actual page
        \begin{titlepage}
            \vspace*{10mm} %%% * is used to give space from top
            \flushright\textbf{\Huge {PBGL CNV-seq Analysis v1.0\\}}
            \vspace{0mm} %%% * is used to give space from top
            \textbf{\Large {A Laboratory Manual\\}}
            \vspace{50mm}
            \textbf{\Large {Anibal E. Morales-Zambrana\\}}
            \vspace{10mm}
            \textbf{\Large {Plant Breeding and Genetics Laboratory\\}}
            \vspace{0mm}
            \textbf{\Large {FAO/IAEA Joint Division\\}}
            \vspace{0mm}
            \textbf{\Large {Seibersdorf, Austria\\}}
	    \vspace{10mm}
            \normalsize Created: March, 2021\\
            \vspace*{0mm}
            \normalsize  Last updated: 04 June 2021
            %% \vfill adds at the bottom
            \vfill
            \small\flushleft {{\textbf {Please note:}} \textit {This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views
expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.}}
        \end{titlepage}
        \pagenumbering{arabic}
        \newcommand{\sectionbreak}{\clearpage}

\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\section{Background}
\label{\detokenize{index:background}}
\sphinxAtStartPar
Copy number variation (CNV) analysis using CNVseq, R, Jupyter Notebooks, Miniconda3, Mamba, and Git.

\sphinxAtStartPar
All the commands run in a Linux terminal are preceded by the \sphinxtitleref{\$} prompt sign. To run a command, copy/past the command without the \sphinxtitleref{\$} sign. Those commands run in a Jupyter Notebook are preceded by the \sphinxtitleref{In {[} {]}:}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.
\end{sphinxadmonition}


\section{Installations \sphinxhyphen{} Virtual Environments and Software Packages}
\label{\detokenize{index:installations-virtual-environments-and-software-packages}}
\sphinxAtStartPar
Before installing any necessary software, it is recommended to check if the computer is running 32\sphinxhyphen{}bit or 64\sphinxhyphen{}bit for downloading Miniconda3. Run the following to verify the system:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} uname \PYGZhy{}m
\end{sphinxVerbatim}


\subsection{Miniconda3 (conda) and Mamba}
\label{\detokenize{index:miniconda3-conda-and-mamba}}
\sphinxAtStartPar
Download the Miniconda3, or simply “conda”, installer:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{https://docs.conda.io/en/latest/miniconda.html\#linux-installers}{Miniconda3 installer for Linux}

\end{itemize}

\sphinxAtStartPar
Run the downloaded installer (for a 64\sphinxhyphen{}bit system):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} bash Miniconda3\PYGZhy{}latest\PYGZhy{}Linux\PYGZhy{}x86\PYGZus{}64.sh
\end{sphinxVerbatim}

\sphinxAtStartPar
Open a new terminal window for conda to take effect. The word \sphinxtitleref{(base)} should appear in front of the computer name in the terminal window, like so:

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{terminal_base_env}.png}
\end{figure}

\begin{figure}[htbp]
\centering

\noindent\sphinxincludegraphics{{terminal_base_env_detail}.png}
\end{figure}

\sphinxAtStartPar
Verify the installation and update conda in new terminal window with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda env list
\PYGZdl{} conda update \PYGZhy{}\PYGZhy{}all
\PYGZdl{} conda upgrade \PYGZhy{}\PYGZhy{}all
\end{sphinxVerbatim}

\sphinxAtStartPar
Install mamba library/package manager that will be used for installing software dependencies of the tool:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda install mamba \PYGZhy{}\PYGZhy{}yes
\end{sphinxVerbatim}


\subsection{Git Installation and Repo Cloning}
\label{\detokenize{index:git-installation-and-repo-cloning}}
\sphinxAtStartPar
Git is required for cloning locally (downloading a copy to your local computer) the PBGL CNVseq Github repository. Git and Github are used for version control of software. It keeps track of development, releases, and issues of a software project.

\sphinxAtStartPar
Install \sphinxstylestrong{git} for cloning the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq} software repository from Github, where the latetest version of the tool resides:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mamba install git \PYGZhy{}\PYGZhy{}yes
\end{sphinxVerbatim}

\sphinxAtStartPar
After the instalation, clone PBGL’s CNVseq repository, \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq}, to the local computer in any desired directory.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git clone https://github.com/pbgl/pbgl\PYGZhy{}cnvseq.git
\end{sphinxVerbatim}

\sphinxAtStartPar
The cloning process will depict the following:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{Cloning} \PYG{n}{into} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pbgl\PYGZhy{}cnvseq}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{.}\PYG{o}{.}\PYG{o}{.}
\PYG{n}{remote}\PYG{p}{:} \PYG{n}{Enumerating} \PYG{n}{objects}\PYG{p}{:} \PYG{l+m+mi}{628}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\PYG{n}{remote}\PYG{p}{:} \PYG{n}{Counting} \PYG{n}{objects}\PYG{p}{:} \PYG{l+m+mi}{100}\PYG{o}{\PYGZpc{}} \PYG{p}{(}\PYG{l+m+mi}{336}\PYG{o}{/}\PYG{l+m+mi}{336}\PYG{p}{)}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\PYG{n}{remote}\PYG{p}{:} \PYG{n}{Compressing} \PYG{n}{objects}\PYG{p}{:} \PYG{l+m+mi}{100}\PYG{o}{\PYGZpc{}} \PYG{p}{(}\PYG{l+m+mi}{239}\PYG{o}{/}\PYG{l+m+mi}{239}\PYG{p}{)}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\PYG{n}{remote}\PYG{p}{:} \PYG{n}{Total} \PYG{l+m+mi}{628} \PYG{p}{(}\PYG{n}{delta} \PYG{l+m+mi}{109}\PYG{p}{)}\PYG{p}{,} \PYG{n}{reused} \PYG{l+m+mi}{292} \PYG{p}{(}\PYG{n}{delta} \PYG{l+m+mi}{78}\PYG{p}{)}\PYG{p}{,} \PYG{n}{pack}\PYG{o}{\PYGZhy{}}\PYG{n}{reused} \PYG{l+m+mi}{292}
\PYG{n}{Receiving} \PYG{n}{objects}\PYG{p}{:} \PYG{l+m+mi}{100}\PYG{o}{\PYGZpc{}} \PYG{p}{(}\PYG{l+m+mi}{628}\PYG{o}{/}\PYG{l+m+mi}{628}\PYG{p}{)}\PYG{p}{,} \PYG{l+m+mf}{11.79} \PYG{n}{MiB} \PYG{o}{|} \PYG{l+m+mf}{7.03} \PYG{n}{MiB}\PYG{o}{/}\PYG{n}{s}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\PYG{n}{Resolving} \PYG{n}{deltas}\PYG{p}{:} \PYG{l+m+mi}{100}\PYG{o}{\PYGZpc{}} \PYG{p}{(}\PYG{l+m+mi}{190}\PYG{o}{/}\PYG{l+m+mi}{190}\PYG{p}{)}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\end{sphinxVerbatim}

\sphinxAtStartPar
The \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq} repository should have been clones successfully. Verify that the download is complete by listing the folders/files in the directory.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} ls \PYGZhy{}l
\end{sphinxVerbatim}

\sphinxAtStartPar
The folder called \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq} should be listed in the directory.


\subsection{Required Libraries with Mamba}
\label{\detokenize{index:required-libraries-with-mamba}}
\sphinxAtStartPar
CNVseq has multiple dependencies, listed below:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Samtools

\item {} 
\sphinxAtStartPar
R
\begin{itemize}
\item {} 
\sphinxAtStartPar
configr

\item {} 
\sphinxAtStartPar
ggplot2

\item {} 
\sphinxAtStartPar
BiocManager

\item {} 
\sphinxAtStartPar
Bioconductor\sphinxhyphen{}GenomicAlignments

\item {} 
\sphinxAtStartPar
Bioconductor\sphinxhyphen{}GenomeInfoDb

\end{itemize}

\item {} 
\sphinxAtStartPar
Jupyter Notebook
\begin{itemize}
\item {} 
\sphinxAtStartPar
IRkernel

\end{itemize}

\end{itemize}

\sphinxAtStartPar
The necessary R packages are installed through the Jupyter Notebook. It proves as a faster and error\sphinxhyphen{}free way to install \sphinxstylestrong{configr}, \sphinxstylestrong{ggplot2}, \sphinxstylestrong{Biocmanager}, \sphinxstylestrong{Bioconductor\sphinxhyphen{}GenomicAlignments}, and \sphinxstylestrong{Bioconductor\sphinxhyphen{}GenomeInfoDb} packages.

\sphinxAtStartPar
There are two ways to install the rest of the necessary libraries to run CNV\sphinxhyphen{}seq: automatically or manually.


\subsubsection{Automatically (faster)}
\label{\detokenize{index:automatically-faster}}
\sphinxAtStartPar
One YAML file, \sphinxstylestrong{environment.yml}, is provided to automatically create a virtual environment and install the dependent libraries through mamba. The file creates the \sphinxstylestrong{cnvseq} virtual environment, along R, Jupyter Notebook, and the R\sphinxhyphen{}kernel in Jupyter. It also installs the dependent R libraries. Run \sphinxstylestrong{environment.yml}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mamba env create \PYGZhy{}\PYGZhy{}file envs/environment.yml
\end{sphinxVerbatim}

\sphinxAtStartPar
Once done, a list of the virtual environments available can be seen by running:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda env list
\end{sphinxVerbatim}

\sphinxAtStartPar
Activate (enter) the recently\sphinxhyphen{}created virtual environment \sphinxstylestrong{cnvseq}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda activate cnvseq
\end{sphinxVerbatim}

\sphinxAtStartPar
Once done, the virtual environment should be activated and all the necessary packages should be installed. This can be verified with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda list
\end{sphinxVerbatim}


\subsubsection{Manually (slower)}
\label{\detokenize{index:manually-slower}}
\sphinxAtStartPar
To manually create and activate an environment, run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda create \PYGZhy{}\PYGZhy{}name cnvseq
\PYGZdl{} conda activate cnvseq
\end{sphinxVerbatim}

\sphinxAtStartPar
Start running the installations of the necessary libraries:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mamba install \PYGZhy{}\PYGZhy{}channel conda\PYGZhy{}forge notebook r\PYGZhy{}irkernel r\PYGZhy{}biocmanager \PYGZhy{}\PYGZhy{}yes
\PYGZdl{} mamba install \PYGZhy{}\PYGZhy{}channel bioconda samtools bioconductor\PYGZhy{}genomicalignments bioconductor\PYGZhy{}genomeinfodb \PYGZhy{}\PYGZhy{}yes
\PYGZdl{} mamba install \PYGZhy{}\PYGZhy{}channel r r\PYGZhy{}ggplot2 \PYGZhy{}\PYGZhy{}yes
\PYGZdl{} mamba install \PYGZhy{}\PYGZhy{}channel pcgr r\PYGZhy{}configr \PYGZhy{}\PYGZhy{}yes
\end{sphinxVerbatim}

\sphinxAtStartPar
Once done, all the necessary packages should be installed. This can be verified with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda list
\end{sphinxVerbatim}


\section{Running Jupyter}
\label{\detokenize{index:running-jupyter}}
\sphinxAtStartPar
To activate Jupyter, run the following in the terminal:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} jupyter notebook
\end{sphinxVerbatim}

\sphinxAtStartPar
This command will start a Jupyter session inside the directory the command is run. The user can navigate between directories, visualize files, and edit files in a web browser by clicking on directories or files, respectively.

\sphinxAtStartPar
Look for the directory \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq} and click on it. Click on \sphinxstylestrong{tool} directory, which contains three directories and two Jupyter Notebooks. Here is a breakdown of each:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{config}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
directory containing configuration files specifying file paths, parameter definitions, and comparison lists

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{helper\sphinxhyphen{}functions}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
directory containing R scripts with functions to calculate and plot CNVs

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{output}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
directory that will contain both tab\sphinxhyphen{}files and images output after running a CNVseq analysis

\end{itemize}

\item {} 
\sphinxAtStartPar
two Jupyter Notebooks:
\begin{itemize}
\item {} 
\sphinxAtStartPar
RCNV\sphinxhyphen{}seq\sphinxhyphen{}PBGL\sphinxhyphen{}sorghum\sphinxhyphen{}analysis\sphinxhyphen{}example.ipynb
\begin{itemize}
\item {} 
\sphinxAtStartPar
example analysis of a comparison between a control and mutant of sorghum

\end{itemize}

\item {} 
\sphinxAtStartPar
RCNV\sphinxhyphen{}seq\sphinxhyphen{}template.ipynb
\begin{itemize}
\item {} 
\sphinxAtStartPar
template for the user

\end{itemize}

\end{itemize}

\end{itemize}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Jupyter lets the user duplicate, rename, move, download, view, or edit files in a web browser. This can be done by clicking the box next to a file and choosing accordingly.
\end{sphinxadmonition}


\subsection{Editing the Configuration File}
\label{\detokenize{index:editing-the-configuration-file}}
\sphinxAtStartPar
In order to run the CNVseq Jupyter Notebook, the user needs to feed it with a configuration file (\sphinxstylestrong{config\sphinxhyphen{}CNVseq.yml}) that specifies the paths to the bam files, comparisons to be done, chromosomes to analyze, and parameter definitions for calculating and plotting CNVs.

\sphinxAtStartPar
The configuration file \sphinxstylestrong{config\sphinxhyphen{}CNVseq.yml} can be found in the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool/config} directory. The configuration file contains the following:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} parameters for CNV/window size calculations defaults from HLiang:}
\PYG{c+c1}{\PYGZsh{} https://github.com/hliang/cnv\PYGZhy{}seq}
\PYG{n}{parameters}\PYG{p}{:}
  \PYG{n}{annotate}\PYG{p}{:} \PYG{n}{TRUE}
  \PYG{n}{bed\PYGZus{}file\PYGZus{}present}\PYG{p}{:} \PYG{n}{FALSE}
  \PYG{n}{bigger}\PYG{p}{:} \PYG{l+m+mf}{1.5}
  \PYG{n}{log2}\PYG{p}{:} \PYG{l+m+mf}{0.6}
  \PYG{n}{pvalue}\PYG{p}{:} \PYG{l+m+mf}{0.001}
  \PYG{n}{window\PYGZus{}size}\PYG{p}{:} \PYG{l+m+mi}{10000}

\PYG{c+c1}{\PYGZsh{} list of chromosomes to be analyzed}
\PYG{n}{chromosomes}\PYG{p}{:}
  \PYG{o}{\PYGZhy{}} \PYG{n}{Chromosome1}
  \PYG{o}{\PYGZhy{}} \PYG{n}{Chromosome2}
  \PYG{o}{\PYGZhy{}} \PYG{n}{Chromosome3}
  \PYG{o}{\PYGZhy{}} \PYG{n}{AnotherChromosome}

\PYG{c+c1}{\PYGZsh{} folder to store outputs; should be inside \PYGZdq{}output\PYGZdq{} directory, example:}
\PYG{c+c1}{\PYGZsh{} output\PYGZus{}path: output/run\PYGZhy{}name or output/organism}
\PYG{n}{output\PYGZus{}path}\PYG{p}{:} \PYG{n}{output}\PYG{o}{/}\PYG{n}{organism}\PYG{o}{\PYGZhy{}}\PYG{n}{being}\PYG{o}{\PYGZhy{}}\PYG{n}{analyzed}

\PYG{c+c1}{\PYGZsh{} path to bed file for varying window sizes (optional)}
\PYG{n}{bed\PYGZus{}path}\PYG{p}{:} \PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{to}\PYG{o}{/}\PYG{n}{bed}\PYG{o}{/}\PYG{n}{file}\PYG{o}{.}\PYG{n}{bed}

\PYG{c+c1}{\PYGZsh{} paths to bam files}
\PYG{n}{paths}\PYG{p}{:}
  \PYG{n}{control}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{control}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1} \PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{to}\PYG{o}{/}\PYG{n}{bam}\PYG{o}{/}\PYG{n}{file}\PYG{o}{/}\PYG{n}{control}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.}\PYG{n}{bam}
  \PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1} \PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{to}\PYG{o}{/}\PYG{n}{bam}\PYG{o}{/}\PYG{n}{file}\PYG{o}{/}\PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.}\PYG{n}{bam}
  \PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2} \PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{to}\PYG{o}{/}\PYG{n}{bam}\PYG{o}{/}\PYG{n}{file}\PYG{o}{/}\PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.}\PYG{n}{bam}

\PYG{c+c1}{\PYGZsh{} comparisons to be analyzed}
\PYG{n}{comparisons}\PYG{p}{:}
  \PYG{n}{control}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{o}{\PYGZhy{}}\PYG{n}{vs}\PYG{o}{\PYGZhy{}}\PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{:}
    \PYG{n}{control}\PYG{p}{:} \PYG{o}{*}\PYG{n}{control}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}
    \PYG{n}{mutant}\PYG{p}{:} \PYG{o}{*}\PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}
  \PYG{n}{control}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{o}{\PYGZhy{}}\PYG{n}{vs}\PYG{o}{\PYGZhy{}}\PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{p}{:}
    \PYG{n}{control}\PYG{p}{:} \PYG{o}{*}\PYG{n}{control}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}
    \PYG{n}{mutant}\PYG{p}{:} \PYG{o}{*}\PYG{n}{mutant}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}
\end{sphinxVerbatim}

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The user needs to edit \sphinxstylestrong{config\sphinxhyphen{}CNVseq.yml} to point towards bam/bed files; specify comparisons and chromosomes to analyze; and define the parameters to calculate/plot CNVs.
\end{sphinxadmonition}

\sphinxAtStartPar
One example configuration files is provided (\sphinxstylestrong{config\sphinxhyphen{}CNVseq\sphinxhyphen{}PBGL\sphinxhyphen{}sorghum\sphinxhyphen{}analysis\sphinxhyphen{}example.yml}). The configuration file \sphinxstylestrong{config\sphinxhyphen{}CNVseq.yml} contains multiple fields to be defined by the user.
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{parameters}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
parameters used to create window sizes, thresholds, plots, etc

\item {} 
\sphinxAtStartPar
the parameter defaults are provided accroding to HLiang’s original values

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{chromosomes}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
list of chromosome names to analyze

\item {} 
\sphinxAtStartPar
chromosome names can be found in a bam file’s header using the following samtools command:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} samtools view \PYGZhy{}h my\PYGZus{}bam\PYGZus{}file.bam | less \PYGZhy{}S
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{output\_path}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
directory that will contain both tab\sphinxhyphen{}files and images output after running a CNVseq analysis

\item {} 
\sphinxAtStartPar
the defined path will be inside the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool/output} directory

\item {} 
\sphinxAtStartPar
\sphinxtitleref{images}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
directory that will store the CNV image outputs per comparison of:

\item {} 
\sphinxAtStartPar
all chromosomes in one plot

\item {} 
\sphinxAtStartPar
each chromosome individually in one plot

\item {} 
\sphinxAtStartPar
any zoomed\sphinxhyphen{}in region plot of one chromosome

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{tab\sphinxhyphen{}files}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
directory that will contain two types of output tab\sphinxhyphen{}delimited files:
\begin{itemize}
\item {} 
\sphinxAtStartPar
hits used to calculate CNVs, containing chromosome, start, end, width, control coverage, mutant coverage

\item {} 
\sphinxAtStartPar
CNVs per chromosome per comparison, containing CNV number, chromosome, start, end, size, log2, and p\sphinxhyphen{}value

\end{itemize}

\end{itemize}

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{bed\_path}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
path pointing to bed file containing targets if using varying window sizes

\item {} 
\sphinxAtStartPar
it is recommended to run the analysis without a \sphinxstylestrong{.bed} file; plots will only reflect the targets in a \sphinxstylestrong{.bed} file

\item {} 
\sphinxAtStartPar
if a \sphinxstylestrong{.bed} file is provided, the \sphinxtitleref{bed\_file\_present} parameter under the \sphinxtitleref{parameters} section has to be changed to \sphinxtitleref{TRUE}

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{paths}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
sample names and their respective paths to \sphinxstylestrong{.bam} files

\item {} 
\sphinxAtStartPar
samples can be named as desired but the sample name must be repeated after the colon and prefixed with a \sphinxtitleref{\&} sign

\item {} 
\sphinxAtStartPar
the \sphinxtitleref{\&} prefix sign is used to reference the sample’s path in different places of the same configuration file

\item {} 
\sphinxAtStartPar
example use:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{paths}\PYG{p}{:}
  \PYG{n}{mysample}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{mysample} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{username}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{mysample}\PYG{o}{.}\PYG{n}{bam}
  \PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{123}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{123} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{username}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{123.}\PYG{n}{bam}
  \PYG{n}{potato95}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{potato95} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{username}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{potato95}\PYG{o}{.}\PYG{n}{bam}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{comparisons}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
comparison names with respective control and mutant samples per comparison

\item {} 
\sphinxAtStartPar
each comparison can be named as desired

\item {} 
\sphinxAtStartPar
the sample names to be used as \sphinxtitleref{control} and \sphinxtitleref{mutant} need to be prefixed by a \sphinxtitleref{*} sign

\item {} 
\sphinxAtStartPar
the \sphinxtitleref{*} prefixed sign is used to extract the sample’s path defined in the \sphinxtitleref{paths} section

\item {} 
\sphinxAtStartPar
example:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{comparisons}\PYG{p}{:}
  \PYG{n}{comparison}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{:}
    \PYG{n}{control}\PYG{p}{:} \PYG{o}{*}\PYG{n}{mysample}
    \PYG{n}{mutant}\PYG{p}{:} \PYG{o}{*}\PYG{n}{potato95}
  \PYG{n}{a}\PYG{o}{\PYGZhy{}}\PYG{n}{different}\PYG{o}{\PYGZhy{}}\PYG{n}{comparison}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{278}\PYG{n}{asd}\PYG{p}{:}
    \PYG{n}{control}\PYG{p}{:} \PYG{o}{*}\PYG{n}{mysample}
    \PYG{n}{mutant}\PYG{p}{:} \PYG{o}{*}\PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{123}
\end{sphinxVerbatim}


\subsection{Running a RCNV\_seq\sphinxhyphen{}template Jupyter Notebook}
\label{\detokenize{index:running-a-rcnv-seq-template-jupyter-notebook}}
\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
It is recommended to duplicate the \sphinxstylestrong{RCNV\sphinxhyphen{}seq\sphinxhyphen{}template} notebook and then renaming the copy before doing any edits to the notebook.
\end{sphinxadmonition}

\sphinxAtStartPar
In the \sphinxstylestrong{pbgl\sphinxhyphen{}cvnseq/tool} directory, click on \sphinxstylestrong{RCNV\sphinxhyphen{}seq\sphinxhyphen{}template} and a new tab in your web\sphinxhyphen{}browser will open the notebook.

\sphinxAtStartPar
The notebook contains cells that are populated by text or code. Instructions are provided in the notebook to guide the user. To run a cell, click on the corresponding cell and click on the \sphinxtitleref{Run} button on the top of the notebook. Another way to run a cell can be done by clicking on the corresponding cell and pressing \sphinxstylestrong{Ctrl + Enter} or \sphinxstylestrong{Shift + Enter}.

\sphinxAtStartPar
The notebook consists of 6 sections:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Installing Required Libraries (optional)

\item {} 
\sphinxAtStartPar
Loading Required Libraries (mandatory)

\item {} 
\sphinxAtStartPar
User Input (mandatory)

\item {} 
\sphinxAtStartPar
CNV Calculations

\item {} 
\sphinxAtStartPar
Plotting

\item {} 
\sphinxAtStartPar
Plotting a Zoomed\sphinxhyphen{}In Region of One Chromosome

\end{enumerate}


\subsubsection{Installing Required Libraries (optional)}
\label{\detokenize{index:installing-required-libraries-optional}}\begin{itemize}
\item {} 
\sphinxAtStartPar
libraries being installed:

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
In [ ]: \PYGZsh{} install necessary libraries using R functions
        if (!requireNamespace(\PYGZdq{}BiocManager\PYGZdq{}, quietly = TRUE))
            install.packages(\PYGZdq{}BiocManager\PYGZdq{})

        BiocManager::install(c(\PYGZdq{}GenomicAlignments\PYGZdq{}, \PYGZdq{}GenomeInfoDb\PYGZdq{}))
        install.packages(\PYGZdq{}configr\PYGZdq{})
        install.packages(\PYGZdq{}ggplot2\PYGZdq{})
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
to be run if the mamba installations were not successful or the loading of the required libraries fails under the \sphinxstylestrong{Loading Required Libraries} section

\end{itemize}


\subsubsection{Loading Required Libraries (mandatory)}
\label{\detokenize{index:loading-required-libraries-mandatory}}\begin{itemize}
\item {} 
\sphinxAtStartPar
libraries being loaded:

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{c+c1}{\PYGZsh{} load necessary libraries}
        \PYG{n}{library}\PYG{p}{(}\PYG{n}{GenomicAlignments}\PYG{p}{)}
        \PYG{n}{library}\PYG{p}{(}\PYG{n}{ggplot2}\PYG{p}{)}
        \PYG{n}{library}\PYG{p}{(}\PYG{n}{configr}\PYG{p}{)}

        \PYG{c+c1}{\PYGZsh{} specify source R script with helper functions}
        \PYG{n}{source}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{helper\PYGZhy{}functions/RCNV\PYGZus{}seq\PYGZhy{}helper.R}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
        \PYG{n}{source}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{helper\PYGZhy{}functions/cnvHLiang.R}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
this cell will load necessary libraries and scripts containing the necessary functions to be used

\item {} 
\sphinxAtStartPar
if running this cell fails, some libraries may be missing from the installation
\begin{itemize}
\item {} 
\sphinxAtStartPar
to fix this issue, run the installations under the \sphinxstylestrong{Installing Required Libraries}

\end{itemize}

\end{itemize}


\subsubsection{User Input (mandatory)}
\label{\detokenize{index:user-input-mandatory}}\begin{itemize}
\item {} 
\sphinxAtStartPar
the user needs to write the appropriate name of the configuration file being used in the following cell:

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{configPath} \PYG{o}{\PYGZlt{}}\PYG{o}{\PYGZhy{}} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{config/config\PYGZhy{}CNVseq.yml}\PYG{l+s+s2}{\PYGZdq{}}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
the bottom cell will extract the fields defined in the configuration file:

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{config} \PYG{o}{\PYGZlt{}}\PYG{o}{\PYGZhy{}} \PYG{n}{read}\PYG{o}{.}\PYG{n}{config}\PYG{p}{(}\PYG{n}{configPath}\PYG{p}{)}
\end{sphinxVerbatim}


\subsubsection{CNV Calculations}
\label{\detokenize{index:cnv-calculations}}\begin{itemize}
\item {} 
\sphinxAtStartPar
function to calculate all the hits and CNVs:

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{cnvCalculate}\PYG{p}{(}\PYG{n}{config}\PYG{p}{)}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
output tabulated files are stored inside \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool/output/name\_defined\_in\_output\_path\_of\_config/tab\sphinxhyphen{}files} directory

\end{itemize}


\subsubsection{Plotting}
\label{\detokenize{index:plotting}}\begin{itemize}
\item {} 
\sphinxAtStartPar
function to plot two types of images:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
CNVs of all chromosomes in the same plot

\item {} 
\sphinxAtStartPar
CNVs of one chromosome per plot

\end{enumerate}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{cnvPlot}\PYG{p}{(}\PYG{n}{config}\PYG{p}{,} \PYG{n}{imgType}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{yMin}\PYG{o}{=} \PYG{p}{,} \PYG{n}{yMax}\PYG{o}{=} \PYG{p}{)}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
function parameters are:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{config} \sphinxhyphen{} configuration file defined under \sphinxstylestrong{User Input} section

\item {} 
\sphinxAtStartPar
\sphinxtitleref{imgType} \sphinxhyphen{}  image extention to use; available options are: \sphinxtitleref{png}, \sphinxtitleref{jpeg}, \sphinxtitleref{svg}, and \sphinxtitleref{pdf}; default to \sphinxtitleref{png}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{yMin} \sphinxhyphen{} y\sphinxhyphen{}axis bottom limit; default to \sphinxtitleref{\sphinxhyphen{}5}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{yMax} \sphinxhyphen{} y\sphinxhyphen{}axis upper limit; default to \sphinxtitleref{5}

\end{itemize}

\item {} 
\sphinxAtStartPar
output images are stored inside the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool/output/name\_defined\_in\_output\_path\_of\_config/images} directory

\item {} 
\sphinxAtStartPar
it is recommended to inspect the CNV\sphinxhyphen{}plots with default y\sphinxhyphen{}limits and then modify

\item {} 
\sphinxAtStartPar
\sphinxtitleref{cnvPlotting} can be used in two ways: with default values or user\sphinxhyphen{}defined values
\begin{itemize}
\item {} 
\sphinxAtStartPar
with default values, the parameters can be omitted and set to \sphinxtitleref{imgType=”png”}, \sphinxtitleref{yMin=\sphinxhyphen{}5} and \sphinxtitleref{yMax=5}

\item {} 
\sphinxAtStartPar
the following two commands have the same parameter values and will output the same plots:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{cnvPlot}\PYG{p}{(}\PYG{n}{config}\PYG{p}{)}
        \PYG{n}{cnvPlot}\PYG{p}{(}\PYG{n}{config}\PYG{p}{,} \PYG{n}{imgType}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{png}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{yMin}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{yMax}\PYG{o}{=}\PYG{l+m+mi}{5}\PYG{p}{)}
\end{sphinxVerbatim}


\subsubsection{Plotting a Zoomed\sphinxhyphen{}In Region of One Chromosome}
\label{\detokenize{index:plotting-a-zoomed-in-region-of-one-chromosome}}\begin{itemize}
\item {} 
\sphinxAtStartPar
function to plot a specific zoomed\sphinxhyphen{}in region of one chromosome

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{cnvPlotZoom}\PYG{p}{(}\PYG{n}{config}\PYG{p}{,} \PYG{n}{tabFile}\PYG{p}{,} \PYG{n}{chromosome}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{start}\PYG{o}{=} \PYG{p}{,} \PYG{n}{end}\PYG{o}{=} \PYG{p}{,} \PYG{n}{yMin}\PYG{o}{=} \PYG{p}{,} \PYG{n}{yMax}\PYG{o}{=} \PYG{p}{,} \PYG{n}{imgType}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
function parameters are:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{config} \sphinxhyphen{} configuration file defined under \sphinxstylestrong{User Input} section

\item {} 
\sphinxAtStartPar
\sphinxtitleref{tabFile} \sphinxhyphen{}  tabulated file containing all hits; requires user\sphinxhyphen{}input to define path to all\sphinxhyphen{}hits tabulated file

\item {} 
\sphinxAtStartPar
\sphinxtitleref{chromosome} \sphinxhyphen{} chromosome name to focus on; default to \sphinxtitleref{NA}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{start} \sphinxhyphen{} start of window in bp; both scientific notation is accepted: \sphinxtitleref{100000} or \sphinxtitleref{10e4}; defaults to \sphinxtitleref{NA}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{end} \sphinxhyphen{} end of window in bp; the same applies as in the \sphinxtitleref{start} parameter; defaults to NA

\item {} 
\sphinxAtStartPar
\sphinxtitleref{yMin} \sphinxhyphen{} y\sphinxhyphen{}axis bottom limit; default to \sphinxtitleref{\sphinxhyphen{}5}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{yMax} \sphinxhyphen{} y\sphinxhyphen{}axis upper limit; default to \sphinxtitleref{5}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{imgType} \sphinxhyphen{}  image extention to use; available options are: \sphinxtitleref{png}, \sphinxtitleref{jpeg}, \sphinxtitleref{svg}, and \sphinxtitleref{pdf}; defaults to \sphinxtitleref{png}

\end{itemize}

\item {} 
\sphinxAtStartPar
requires path definition of \sphinxtitleref{tabFile} parameter in the cell:

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{tabFile} \PYG{o}{\PYGZlt{}}\PYG{o}{\PYGZhy{}} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{output/run\PYGZhy{}name/tab\PYGZhy{}files/tab\PYGZhy{}file.tab}\PYG{l+s+s2}{\PYGZdq{}}
\end{sphinxVerbatim}


\section{Example Tutorial \sphinxhyphen{} CNV Analysis of Sorghum}
\label{\detokenize{index:example-tutorial-cnv-analysis-of-sorghum}}
\sphinxAtStartPar
In this section of the manual, an example analysis of sorghum will be shown in a step\sphinxhyphen{}by\sphinxhyphen{}step process. The data has been subset to three chromosomes Chr04, Chr05, and Chr09. The CNVseq analysis will depict a deletion on chromosome Chr09.

\sphinxAtStartPar
The tutorial is divided between the following sections:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Data Download

\item {} 
\sphinxAtStartPar
General Installations

\item {} 
\sphinxAtStartPar
Github Repository Cloning

\item {} 
\sphinxAtStartPar
Virtual Environment Creation

\item {} 
\sphinxAtStartPar
Configuration File Editing

\item {} 
\sphinxAtStartPar
Jupyter Notebook Analysis

\end{enumerate}


\subsection{Data Download}
\label{\detokenize{index:data-download}}
\sphinxAtStartPar
Running this tutorial requires two binary alignment map (BAM) files of sorghum crop: one control and one mutant. Each \sphinxstylestrong{.bam} file is subset for chromosomes Chr04, Chr05, and Chr09; this is reflected on their names. They have been both sorted and indexed using samtools.

\sphinxAtStartPar
The following links can be used to download the necessary \sphinxstylestrong{.bam} and \sphinxstylestrong{.bam.bai} (index) files:
\begin{itemize}
\item {} 
\sphinxAtStartPar
con\sphinxhyphen{}2\_S1 (sorghum control)
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxurl{https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021\_Training/con-2\_S1-Chromes-04-05-09.bam}

\item {} 
\sphinxAtStartPar
\sphinxurl{https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021\_Training/con-2\_S1-Chromes-04-05-09.bam.bai}

\end{itemize}

\item {} 
\sphinxAtStartPar
D2\sphinxhyphen{}1\_S7 (sorghum mutant)
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxurl{https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021\_Training/D2-1\_S7-Chromes-04-05-09.bam}

\item {} 
\sphinxAtStartPar
\sphinxurl{https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021\_Training/D2-1\_S7-Chromes-04-05-09.bam.bai}

\end{itemize}

\end{itemize}

\sphinxAtStartPar
There are two additional ways to download the \sphinxstylestrong{.bam} files and their respectives indices, besides clicking the links above:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Linux terminal

\item {} 
\sphinxAtStartPar
Web\sphinxhyphen{}Browser

\end{enumerate}


\subsubsection{Linux Terminal}
\label{\detokenize{index:linux-terminal}}
\sphinxAtStartPar
Open a new terminal and navigate to a directory of choice. We recommend creating a directory to store the data and running \sphinxtitleref{wget} in the respective location:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mkdir bam\PYGZus{}files
\PYGZdl{} cd bam\PYGZus{}files
\PYGZdl{} wget https://bss1innov1nafa1poc1.blob.core.windows.net/sample\PYGZhy{}container/2021\PYGZus{}Training/con\PYGZhy{}2\PYGZus{}S1\PYGZhy{}Chromes\PYGZhy{}04\PYGZhy{}05\PYGZhy{}09.bam
\PYGZdl{} wget https://bss1innov1nafa1poc1.blob.core.windows.net/sample\PYGZhy{}container/2021\PYGZus{}Training/con\PYGZhy{}2\PYGZus{}S1\PYGZhy{}Chromes\PYGZhy{}04\PYGZhy{}05\PYGZhy{}09.bam.bai
\PYGZdl{} wget https://bss1innov1nafa1poc1.blob.core.windows.net/sample\PYGZhy{}container/2021\PYGZus{}Training/D2\PYGZhy{}1\PYGZus{}S7\PYGZhy{}Chromes\PYGZhy{}04\PYGZhy{}05\PYGZhy{}09.bam
\PYGZdl{} wget https://bss1innov1nafa1poc1.blob.core.windows.net/sample\PYGZhy{}container/2021\PYGZus{}Training/D2\PYGZhy{}1\PYGZus{}S7\PYGZhy{}Chromes\PYGZhy{}04\PYGZhy{}05\PYGZhy{}09.bam.bai
\end{sphinxVerbatim}


\subsubsection{Web\sphinxhyphen{}Browser}
\label{\detokenize{index:web-browser}}
\sphinxAtStartPar
Open a web\sphinxhyphen{}browser of preference. Copy/paste the links provided above in the address bar. This should automatically begin the download. Move the downloaded files to a location of personal preference.


\subsection{General Installations}
\label{\detokenize{index:general-installations}}
\sphinxAtStartPar
Open a web browser and copy/paste the following link to download Miniconda3:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxurl{https://docs.conda.io/en/latest/miniconda.html\#linux-installers}

\end{itemize}

\sphinxAtStartPar
After download, open a new terminal window, navigate to the directory with the downloaded Miniconda3 installer, and run the installation.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The donwloaded Miniconda3 installer file might not match the one run in this example. Please, type the corresponding name of the \sphinxstylestrong{.sh} file downloaded.
\end{sphinxadmonition}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} bash Miniconda3\PYGZhy{}latest\PYGZhy{}Linux\PYGZhy{}x86\PYGZus{}64.sh
\end{sphinxVerbatim}

\sphinxAtStartPar
Once the Miniconda3 installation is done, close the terminal and open a new one. The word \sphinxtitleref{(base)} should be present to the left of the computer name in the prompt. Update conda and install mamba.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda update \PYGZhy{}\PYGZhy{}all \PYGZhy{}\PYGZhy{}yes
\PYGZdl{} conda upgrade \PYGZhy{}\PYGZhy{}all \PYGZhy{}\PYGZhy{}yes
\PYGZdl{} conda install mamba \PYGZhy{}\PYGZhy{}yes
\end{sphinxVerbatim}


\subsection{Github Repository Cloning}
\label{\detokenize{index:github-repository-cloning}}
\sphinxAtStartPar
Git or Github is used for storage and version control of software projects. Git is used to manage in a local Linux terminal. First, git will need to be installed. In a new terminal window, run the installation command:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mamba install git \PYGZhy{}\PYGZhy{}yes
\end{sphinxVerbatim}

\sphinxAtStartPar
After installing git, create a location to store Github repoitories, navigate into it, and clone (download a copy locally) of PBGL’s \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq} repository.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mkdir Github
\PYGZdl{} cd Github
\PYGZdl{} git clone https://github.com/pbgl/pbgl\PYGZhy{}cnvseq
\end{sphinxVerbatim}

\sphinxAtStartPar
A successful cloning will output the following:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{Cloning} \PYG{n}{into} \PYG{l+s+s1}{\PYGZsq{}}\PYG{l+s+s1}{pbgl\PYGZhy{}cnvseq}\PYG{l+s+s1}{\PYGZsq{}}\PYG{o}{.}\PYG{o}{.}\PYG{o}{.}
\PYG{n}{remote}\PYG{p}{:} \PYG{n}{Enumerating} \PYG{n}{objects}\PYG{p}{:} \PYG{l+m+mi}{628}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\PYG{n}{remote}\PYG{p}{:} \PYG{n}{Counting} \PYG{n}{objects}\PYG{p}{:} \PYG{l+m+mi}{100}\PYG{o}{\PYGZpc{}} \PYG{p}{(}\PYG{l+m+mi}{336}\PYG{o}{/}\PYG{l+m+mi}{336}\PYG{p}{)}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\PYG{n}{remote}\PYG{p}{:} \PYG{n}{Compressing} \PYG{n}{objects}\PYG{p}{:} \PYG{l+m+mi}{100}\PYG{o}{\PYGZpc{}} \PYG{p}{(}\PYG{l+m+mi}{239}\PYG{o}{/}\PYG{l+m+mi}{239}\PYG{p}{)}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\PYG{n}{remote}\PYG{p}{:} \PYG{n}{Total} \PYG{l+m+mi}{628} \PYG{p}{(}\PYG{n}{delta} \PYG{l+m+mi}{109}\PYG{p}{)}\PYG{p}{,} \PYG{n}{reused} \PYG{l+m+mi}{292} \PYG{p}{(}\PYG{n}{delta} \PYG{l+m+mi}{78}\PYG{p}{)}\PYG{p}{,} \PYG{n}{pack}\PYG{o}{\PYGZhy{}}\PYG{n}{reused} \PYG{l+m+mi}{292}
\PYG{n}{Receiving} \PYG{n}{objects}\PYG{p}{:} \PYG{l+m+mi}{100}\PYG{o}{\PYGZpc{}} \PYG{p}{(}\PYG{l+m+mi}{628}\PYG{o}{/}\PYG{l+m+mi}{628}\PYG{p}{)}\PYG{p}{,} \PYG{l+m+mf}{11.79} \PYG{n}{MiB} \PYG{o}{|} \PYG{l+m+mf}{7.03} \PYG{n}{MiB}\PYG{o}{/}\PYG{n}{s}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\PYG{n}{Resolving} \PYG{n}{deltas}\PYG{p}{:} \PYG{l+m+mi}{100}\PYG{o}{\PYGZpc{}} \PYG{p}{(}\PYG{l+m+mi}{190}\PYG{o}{/}\PYG{l+m+mi}{190}\PYG{p}{)}\PYG{p}{,} \PYG{n}{done}\PYG{o}{.}
\end{sphinxVerbatim}

\sphinxAtStartPar
Navigate into the cloned repository,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} cd pbgl\PYGZhy{}cnvseq
\end{sphinxVerbatim}


\subsection{Virtual Environment Creation}
\label{\detokenize{index:virtual-environment-creation}}
\sphinxAtStartPar
Once inside the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq} directory, create the \sphinxstylestrong{cnvseq} virtual environment, install the necessary libraries, and activate the newly created \sphinxstylestrong{cnvseq} virtual environment.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} mamba env create \PYGZhy{}\PYGZhy{}file envs/environment.yml
\PYGZdl{} conda activate cnvseq
\end{sphinxVerbatim}

\sphinxAtStartPar
The name \sphinxtitleref{(cnvseq)} environment should be reflected to the left of the computer name in the terminal command prompt.


\subsection{Configuration File Editing}
\label{\detokenize{index:configuration-file-editing}}
\sphinxAtStartPar
Open a Jupyter session by running,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} jupyter notebook
\end{sphinxVerbatim}

\sphinxAtStartPar
Navigate to \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool/config} and click on \sphinxstylestrong{config\sphinxhyphen{}CNVseq\sphinxhyphen{}PBGL\sphinxhyphen{}sorghum\sphinxhyphen{}analysis\sphinxhyphen{}example.yml}. This will open the configuration file in a new web\sphinxhyphen{}browser tab. Copy/paste the following in the configuration file.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The paths to the bam files will not match. They need to be edited accordingly to point towards the location of the bam files stored locally in the user’s computer.
\end{sphinxadmonition}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{c+c1}{\PYGZsh{} parameters for CNV/window size calculations defaults from HLiang:}
\PYG{c+c1}{\PYGZsh{} https://github.com/hliang/cnv\PYGZhy{}seq}
\PYG{n}{parameters}\PYG{p}{:}
  \PYG{n}{annotate}\PYG{p}{:} \PYG{n}{TRUE}
  \PYG{n}{bed\PYGZus{}file\PYGZus{}present}\PYG{p}{:} \PYG{n}{FALSE}
  \PYG{n}{bigger}\PYG{p}{:} \PYG{l+m+mf}{1.5}
  \PYG{n}{log2}\PYG{p}{:} \PYG{l+m+mf}{0.6}
  \PYG{n}{pvalue}\PYG{p}{:} \PYG{l+m+mf}{0.001}
  \PYG{n}{window\PYGZus{}size}\PYG{p}{:} \PYG{l+m+mi}{10000}

\PYG{c+c1}{\PYGZsh{} list of chromosomes to be analyzed}
\PYG{n}{chromosomes}\PYG{p}{:}
  \PYG{o}{\PYGZhy{}} \PYG{n}{Chr04}
  \PYG{o}{\PYGZhy{}} \PYG{n}{Chr05}
  \PYG{o}{\PYGZhy{}} \PYG{n}{Chr09}

\PYG{c+c1}{\PYGZsh{} folder to store outputs; should be inside \PYGZdq{}output\PYGZdq{} directory, example:}
\PYG{c+c1}{\PYGZsh{} output\PYGZus{}path: output/run\PYGZhy{}name or output/organism}
\PYG{n}{output\PYGZus{}path}\PYG{p}{:} \PYG{n}{output}\PYG{o}{/}\PYG{n}{PBGL}\PYG{o}{\PYGZhy{}}\PYG{n}{sorghum}\PYG{o}{\PYGZhy{}}\PYG{n}{analysis}\PYG{o}{\PYGZhy{}}\PYG{n}{example}

\PYG{c+c1}{\PYGZsh{} path to bed file for varying window sizes (optional)}
\PYG{n}{bed\PYGZus{}path}\PYG{p}{:}

\PYG{c+c1}{\PYGZsh{} paths to bam files}
\PYG{n}{paths}\PYG{p}{:}
  \PYG{n}{con}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{n}{\PYGZus{}S1}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{con}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{n}{\PYGZus{}S1} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{anibal}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{sorghum}\PYG{o}{/}\PYG{n}{con}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{n}{\PYGZus{}S1}\PYG{o}{\PYGZhy{}}\PYG{n}{Chromes}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{04}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{05}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{09.}\PYG{n}{bam}
  \PYG{n}{D2}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{n}{\PYGZus{}S7}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{D2}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{n}{\PYGZus{}S7} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{anibal}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{sorghum}\PYG{o}{/}\PYG{n}{D2}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{n}{\PYGZus{}S7}\PYG{o}{\PYGZhy{}}\PYG{n}{Chromes}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{04}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{05}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{09.}\PYG{n}{bam}

\PYG{c+c1}{\PYGZsh{} comparisons to be analyzed}
\PYG{n}{comparisons}\PYG{p}{:}
  \PYG{n}{con}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{n}{\PYGZus{}s1}\PYG{o}{\PYGZhy{}}\PYG{n}{vs}\PYG{o}{\PYGZhy{}}\PYG{n}{D2}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{n}{\PYGZus{}S7}\PYG{p}{:}
    \PYG{n}{control}\PYG{p}{:} \PYG{o}{*}\PYG{n}{con}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{n}{\PYGZus{}S1}
    \PYG{n}{mutant}\PYG{p}{:} \PYG{o}{*}\PYG{n}{D2}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{n}{\PYGZus{}S7}
\end{sphinxVerbatim}

\sphinxAtStartPar
Save the file and close the tab.


\subsection{Jupyter Notebook Analysis}
\label{\detokenize{index:jupyter-notebook-analysis}}
\sphinxAtStartPar
In the open Jupyter session, navigate to the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool} directory and click on the \sphinxstylestrong{RCNV\sphinxhyphen{}seq\sphinxhyphen{}template.ipynb} Jupyter Notebook. Begin by running the \sphinxstylestrong{Loading Required Libraries} section. This can be done by clicking on the cell to be run followed by clicking the \sphinxtitleref{Run} button on the top of the Jupyter Notebook.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:}  \PYG{c+c1}{\PYGZsh{} load necessary libraries}
         \PYG{n}{library}\PYG{p}{(}\PYG{n}{GenomicAlignments}\PYG{p}{)}
         \PYG{n}{library}\PYG{p}{(}\PYG{n}{ggplot2}\PYG{p}{)}
         \PYG{n}{library}\PYG{p}{(}\PYG{n}{configr}\PYG{p}{)}

         \PYG{c+c1}{\PYGZsh{} specify source R script with helper functions}
         \PYG{n}{source}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{helper\PYGZhy{}functions/RCNV\PYGZus{}seq\PYGZhy{}helper.R}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
         \PYG{n}{source}\PYG{p}{(}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{helper\PYGZhy{}functions/cnvHLiang.R}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
If loading the libraries is unsuccessful (specified in the cell output), an attempt to install the missing libraries can be done by running the \sphinxstylestrong{Installing Required Libraries} section,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
In [ ]:  \PYGZsh{} install necessary libraries using R functions
         if (!requireNamespace(\PYGZdq{}BiocManager\PYGZdq{}, quietly = TRUE))
             install.packages(\PYGZdq{}BiocManager\PYGZdq{})

         BiocManager::install(c(\PYGZdq{}GenomicAlignments\PYGZdq{}, \PYGZdq{}GenomeInfoDb\PYGZdq{}))
         install.packages(\PYGZdq{}configr\PYGZdq{})
         install.packages(\PYGZdq{}ggplot2\PYGZdq{})
\end{sphinxVerbatim}

\sphinxAtStartPar
Run the \sphinxstylestrong{User Input} section after correctly editing the name/path of the configuration file:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{configPath} \PYG{o}{\PYGZlt{}}\PYG{o}{\PYGZhy{}} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{config/config\PYGZhy{}CNVseq.yml}\PYG{l+s+s2}{\PYGZdq{}}
\end{sphinxVerbatim}

\sphinxAtStartPar
Load the contents of the configuration file by running the cell,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{config} \PYG{o}{\PYGZlt{}}\PYG{o}{\PYGZhy{}} \PYG{n}{read}\PYG{o}{.}\PYG{n}{config}\PYG{p}{(}\PYG{n}{configPath}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
Copy number variants (CNVs) are calculated under the \sphinxstylestrong{CNV Calculations} section by running the cell,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{cnvCalculate}\PYG{p}{(}\PYG{n}{config}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The output tabulated files of all\sphinxhyphen{}hits and CNVs can be inspected in the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool/output/PBGL\sphinxhyphen{}sorghum\sphinxhyphen{}analysis\sphinxhyphen{}example/tab\sphinxhyphen{}files} directory. These can be used to import data to other softwares of choice for downstream data analysis. A successful run has the following output right under the input cell,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{Comparison}\PYG{p}{:} \PYG{n}{con}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{n}{\PYGZus{}s1}\PYG{o}{\PYGZhy{}}\PYG{n}{vs}\PYG{o}{\PYGZhy{}}\PYG{n}{D2}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{n}{\PYGZus{}S7}
\PYG{n}{Comparing} \PYG{n}{samples}\PYG{p}{:}
\PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{anibal}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{sorghum}\PYG{o}{/}\PYG{n}{con}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{2}\PYG{n}{\PYGZus{}S1}\PYG{o}{\PYGZhy{}}\PYG{n}{Chromes}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{04}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{05}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{09.}\PYG{n}{bam}
\PYG{n}{vs}\PYG{o}{.}
 \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{anibal}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{sorghum}\PYG{o}{/}\PYG{n}{D2}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{n}{\PYGZus{}S7}\PYG{o}{\PYGZhy{}}\PYG{n}{Chromes}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{04}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{05}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{09.}\PYG{n}{bam}

\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chromosome:  Chr04}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chromosome:  Chr05}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{chromosome:  Chr09}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  1  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  2  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  3  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  4  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  5  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  6  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  7  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  8  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  9  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  10  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  11  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  12  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  13  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  14  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  15  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  16  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  17  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  18  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  19  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  20  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  21  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  22  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  23  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  24  of  25}\PYG{l+s+s2}{\PYGZdq{}}
\PYG{p}{[}\PYG{l+m+mi}{1}\PYG{p}{]} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{cnv\PYGZus{}id:  25  of  25}\PYG{l+s+s2}{\PYGZdq{}}

\PYG{n}{CNV} \PYG{n}{percentage} \PYG{o+ow}{in} \PYG{n}{genome}\PYG{p}{:} \PYG{l+m+mf}{1.6}\PYG{o}{\PYGZpc{}}
\PYG{n}{CNV} \PYG{n}{nucleotide} \PYG{n}{content}\PYG{p}{:} \PYG{l+m+mi}{3140000}
\PYG{n}{CNV} \PYG{n}{count}\PYG{p}{:} \PYG{l+m+mi}{25}
\PYG{n}{Mean} \PYG{n}{size}\PYG{p}{:} \PYG{l+m+mi}{125600}
\PYG{n}{Median} \PYG{n}{size}\PYG{p}{:} \PYG{l+m+mi}{85000}
\PYG{n}{Max} \PYG{n}{Size}\PYG{p}{:} \PYG{l+m+mi}{750000}
\PYG{n}{Min} \PYG{n}{Size}\PYG{p}{:} \PYG{l+m+mi}{20000}

\PYG{n}{cnv} \PYG{n}{chromosome}      \PYG{n}{start}   \PYG{n}{end}     \PYG{n}{size}    \PYG{n}{log2}    \PYG{n}{p}\PYG{o}{.}\PYG{n}{value}
\PYG{n}{CNVR\PYGZus{}1}      \PYG{n}{Chr05}   \PYG{l+m+mi}{66247501}        \PYG{l+m+mi}{66267500}        \PYG{l+m+mi}{20000}   \PYG{l+m+mf}{1.013742}        \PYG{l+m+mf}{1.222196e\PYGZhy{}59}
\PYG{n}{CNVR\PYGZus{}2}      \PYG{n}{Chr09}   \PYG{l+m+mi}{23497501}        \PYG{l+m+mi}{23527500}        \PYG{l+m+mi}{30000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.288917}       \PYG{l+m+mf}{3.556107e\PYGZhy{}238}
\PYG{n}{CNVR\PYGZus{}3}      \PYG{n}{Chr09}   \PYG{l+m+mi}{23547501}        \PYG{l+m+mi}{23917500}        \PYG{l+m+mi}{370000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.684822}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}4}      \PYG{n}{Chr09}   \PYG{l+m+mi}{23927501}        \PYG{l+m+mi}{24132500}        \PYG{l+m+mi}{205000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.639143}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}5}      \PYG{n}{Chr09}   \PYG{l+m+mi}{24142501}        \PYG{l+m+mi}{24302500}        \PYG{l+m+mi}{160000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.359219}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}6}      \PYG{n}{Chr09}   \PYG{l+m+mi}{24337501}        \PYG{l+m+mi}{24417500}        \PYG{l+m+mi}{80000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.907744}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}7}      \PYG{n}{Chr09}   \PYG{l+m+mi}{24437501}        \PYG{l+m+mi}{24467500}        \PYG{l+m+mi}{30000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{0.912225}       \PYG{l+m+mf}{1.86383e\PYGZhy{}68}
\PYG{n}{CNVR\PYGZus{}8}      \PYG{n}{Chr09}   \PYG{l+m+mi}{24512501}        \PYG{l+m+mi}{24652500}        \PYG{l+m+mi}{140000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.799081}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}9}      \PYG{n}{Chr09}   \PYG{l+m+mi}{24657501}        \PYG{l+m+mi}{24692500}        \PYG{l+m+mi}{35000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.466217}       \PYG{l+m+mf}{9.479478e\PYGZhy{}297}
\PYG{n}{CNVR\PYGZus{}10}     \PYG{n}{Chr09}   \PYG{l+m+mi}{24747501}        \PYG{l+m+mi}{24887500}        \PYG{l+m+mi}{140000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.46515}        \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}11}     \PYG{n}{Chr09}   \PYG{l+m+mi}{24917501}        \PYG{l+m+mi}{24942500}        \PYG{l+m+mi}{25000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.980229}       \PYG{l+m+mf}{9.759448e\PYGZhy{}172}
\PYG{n}{CNVR\PYGZus{}12}     \PYG{n}{Chr09}   \PYG{l+m+mi}{24947501}        \PYG{l+m+mi}{25007500}        \PYG{l+m+mi}{60000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.378318}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}13}     \PYG{n}{Chr09}   \PYG{l+m+mi}{25017501}        \PYG{l+m+mi}{25102500}        \PYG{l+m+mi}{85000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.464295}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}14}     \PYG{n}{Chr09}   \PYG{l+m+mi}{25107501}        \PYG{l+m+mi}{25132500}        \PYG{l+m+mi}{25000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.254819}       \PYG{l+m+mf}{4.908193e\PYGZhy{}196}
\PYG{n}{CNVR\PYGZus{}15}     \PYG{n}{Chr09}   \PYG{l+m+mi}{25137501}        \PYG{l+m+mi}{25202500}        \PYG{l+m+mi}{65000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.90878}        \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}16}     \PYG{n}{Chr09}   \PYG{l+m+mi}{25212501}        \PYG{l+m+mi}{25422500}        \PYG{l+m+mi}{210000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{3.111295}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}17}     \PYG{n}{Chr09}   \PYG{l+m+mi}{25427501}        \PYG{l+m+mi}{25512500}        \PYG{l+m+mi}{85000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.883919}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}18}     \PYG{n}{Chr09}   \PYG{l+m+mi}{25522501}        \PYG{l+m+mi}{25637500}        \PYG{l+m+mi}{115000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.617425}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}19}     \PYG{n}{Chr09}   \PYG{l+m+mi}{25647501}        \PYG{l+m+mi}{25722500}        \PYG{l+m+mi}{75000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.367854}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}20}     \PYG{n}{Chr09}   \PYG{l+m+mi}{25727501}        \PYG{l+m+mi}{26477500}        \PYG{l+m+mi}{750000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.388962}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}21}     \PYG{n}{Chr09}   \PYG{l+m+mi}{26482501}        \PYG{l+m+mi}{26507500}        \PYG{l+m+mi}{25000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.727876}       \PYG{l+m+mf}{9.894624e\PYGZhy{}147}
\PYG{n}{CNVR\PYGZus{}22}     \PYG{n}{Chr09}   \PYG{l+m+mi}{26512501}        \PYG{l+m+mi}{26552500}        \PYG{l+m+mi}{40000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.624917}       \PYG{l+m+mf}{3.444569e\PYGZhy{}216}
\PYG{n}{CNVR\PYGZus{}23}     \PYG{n}{Chr09}   \PYG{l+m+mi}{26562501}        \PYG{l+m+mi}{26787500}        \PYG{l+m+mi}{225000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{2.290655}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}24}     \PYG{n}{Chr09}   \PYG{l+m+mi}{26817501}        \PYG{l+m+mi}{26937500}        \PYG{l+m+mi}{120000}  \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.704837}       \PYG{l+m+mi}{0}
\PYG{n}{CNVR\PYGZus{}25}     \PYG{n}{Chr09}   \PYG{l+m+mi}{26947501}        \PYG{l+m+mi}{26972500}        \PYG{l+m+mi}{25000}   \PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{1.303558}       \PYG{l+m+mf}{1.960969e\PYGZhy{}100}
\end{sphinxVerbatim}

\sphinxAtStartPar
The output tabulated files under \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool/output/PBGL\sphinxhyphen{}sorghum\sphinxhyphen{}analysis\sphinxhyphen{}example/tab\sphinxhyphen{}files} are the following:
\begin{itemize}
\item {} 
\sphinxAtStartPar
con\sphinxhyphen{}2\_s1\sphinxhyphen{}vs\sphinxhyphen{}D2\sphinxhyphen{}1\_S7\sphinxhyphen{}all\sphinxhyphen{}chroms\sphinxhyphen{}CNVs.tab

\item {} 
\sphinxAtStartPar
con\sphinxhyphen{}2\_s1\sphinxhyphen{}vs\sphinxhyphen{}D2\sphinxhyphen{}1\_S7\sphinxhyphen{}window\sphinxhyphen{}10000\sphinxhyphen{}all\sphinxhyphen{}hits.tab

\end{itemize}

\sphinxAtStartPar
A depiction of the files are shown below,

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{all-hits-tab-file}.png}
\caption{Tabulated file containing all hits of all chromosomes}\label{\detokenize{index:id1}}\end{figure}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{cnvs-tab-file}.png}
\caption{Tabulated file containing all CNVs of all chromsomes}\label{\detokenize{index:id2}}\end{figure}

\sphinxAtStartPar
After the CNV calculations are done, plot the CNVs under the \sphinxstylestrong{Plotting} section. Here, the parameters will be set to the following,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{cnvPlot}\PYG{p}{(}\PYG{n}{confg}\PYG{p}{,} \PYG{n}{imgType}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{png}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{yMin}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{5}\PYG{p}{,} \PYG{n}{yMax}\PYG{o}{=}\PYG{l+m+mi}{5}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The output images under \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool/output/PBGL\sphinxhyphen{}sorghum\sphinxhyphen{}analysis\sphinxhyphen{}example/images} are the following:
\begin{itemize}
\item {} 
\sphinxAtStartPar
con\sphinxhyphen{}2\_s1\sphinxhyphen{}vs\sphinxhyphen{}D2\sphinxhyphen{}1\_S7\sphinxhyphen{}chromosome\sphinxhyphen{}all\sphinxhyphen{}window\sphinxhyphen{}10000.png

\item {} 
\sphinxAtStartPar
con\sphinxhyphen{}2\_s1\sphinxhyphen{}vs\sphinxhyphen{}D2\sphinxhyphen{}1\_S7\sphinxhyphen{}chromosome\sphinxhyphen{}Chr04\sphinxhyphen{}window\sphinxhyphen{}10000.png

\item {} 
\sphinxAtStartPar
con\sphinxhyphen{}2\_s1\sphinxhyphen{}vs\sphinxhyphen{}D2\sphinxhyphen{}1\_S7\sphinxhyphen{}chromosome\sphinxhyphen{}Chr05\sphinxhyphen{}window\sphinxhyphen{}10000.png

\item {} 
\sphinxAtStartPar
con\sphinxhyphen{}2\_s1\sphinxhyphen{}vs\sphinxhyphen{}D2\sphinxhyphen{}1\_S7\sphinxhyphen{}chromosome\sphinxhyphen{}Chr09\sphinxhyphen{}window\sphinxhyphen{}10000.png

\end{itemize}

\sphinxAtStartPar
The output plots are the following,

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{con-2_s1-vs-D2-1_S7-chromosome-all-window-10000}.png}
\caption{CNV plot of all chromosomes}\label{\detokenize{index:id3}}\end{figure}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{con-2_s1-vs-D2-1_S7-chromosome-Chr04-window-10000}.png}
\caption{CNV plot of chromosome Chr04}\label{\detokenize{index:id4}}\end{figure}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{con-2_s1-vs-D2-1_S7-chromosome-Chr05-window-10000}.png}
\caption{CNV plot of chromosome Chr05}\label{\detokenize{index:id5}}\end{figure}

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{con-2_s1-vs-D2-1_S7-chromosome-Chr09-window-10000}.png}
\caption{CNV plot of chromosome Chr09}\label{\detokenize{index:id6}}\end{figure}

\sphinxAtStartPar
As seen above, each plot contains:
\begin{itemize}
\item {} 
\sphinxAtStartPar
a title of the chromosome being analyzed

\item {} 
\sphinxAtStartPar
a sub\sphinxhyphen{}title of the comparison being done

\item {} 
\sphinxAtStartPar
Log2 Ratio in the y\sphinxhyphen{}axis

\item {} 
\sphinxAtStartPar
chromosome name on the x\sphinxhyphen{}axis in plot of all chromosomes

\item {} 
\sphinxAtStartPar
base\sphinxhyphen{}pair (bp) position on the x\sphinxhyphen{}axis in plot of one chromosome

\item {} 
\sphinxAtStartPar
a label under the x\sphinxhyphen{}axis specifying the location of the stored image

\end{itemize}

\sphinxAtStartPar
The CNV plot of chromosome Chr09 shows an artifact. Moreover, this artifact seems to lie between base\sphinxhyphen{}pairs 2e07 and 3e07. To zoom in, the \sphinxstylestrong{Plotting a Zoomed\sphinxhyphen{}In Region of One Chromosome} section is provided.

\sphinxAtStartPar
Under the \sphinxstylestrong{Plotting a Zoomed\sphinxhyphen{}In Region of One Chromosome} section, first specify the location of the all\sphinxhyphen{}hits tabulated file of chromosome Chr09,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{tabFile} \PYG{o}{\PYGZlt{}}\PYG{o}{\PYGZhy{}} \PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{output/PBGL\PYGZhy{}sorghum\PYGZhy{}analysis\PYGZhy{}example/tab\PYGZhy{}files/con\PYGZhy{}2\PYGZus{}s1\PYGZhy{}vs\PYGZhy{}D2\PYGZhy{}1\PYGZus{}S7\PYGZhy{}window\PYGZhy{}10000\PYGZhy{}all\PYGZhy{}hits.tab}\PYG{l+s+s2}{\PYGZdq{}}
\end{sphinxVerbatim}

\sphinxAtStartPar
The run the zooming\sphinxhyphen{}in function with the following parameters,

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{In} \PYG{p}{[} \PYG{p}{]}\PYG{p}{:} \PYG{n}{cnvPlotZoom}\PYG{p}{(}\PYG{n}{config}\PYG{p}{,} \PYG{n}{tabFile}\PYG{p}{,} \PYG{n}{chromosome}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{Chr09}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{,} \PYG{n}{start}\PYG{o}{=}\PYG{l+m+mf}{2.3e7} \PYG{p}{,} \PYG{n}{end}\PYG{o}{=}\PYG{l+m+mf}{2.8e7} \PYG{p}{,} \PYG{n}{yMin}\PYG{o}{=}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{5} \PYG{p}{,} \PYG{n}{yMax}\PYG{o}{=}\PYG{l+m+mf}{2.5}\PYG{p}{,} \PYG{n}{imgType}\PYG{o}{=}\PYG{l+s+s2}{\PYGZdq{}}\PYG{l+s+s2}{png}\PYG{l+s+s2}{\PYGZdq{}}\PYG{p}{)}
\end{sphinxVerbatim}

\sphinxAtStartPar
The output plot, which is also stored in the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvseq/tool/output/PBGL\sphinxhyphen{}sorghum\sphinxhyphen{}analysis\sphinxhyphen{}example/images}, shows a zoomed\sphinxhyphen{}in region of chromosome Chr09 of sorghum. The output path is also specified under the x\sphinxhyphen{}axis label.

\begin{figure}[htbp]
\centering
\capstart

\noindent\sphinxincludegraphics{{con-2_s1-vs-D2-1_S7-window-10000-chromosome-Chr09-zoomed}.png}
\caption{CNV plot of a zoomed\sphinxhyphen{}in region of chromosome Chr09}\label{\detokenize{index:id7}}\end{figure}

\sphinxAtStartPar
Knowing where artifacts like this one is located plays a major role in identifying insertions or deletions in an organism. The user can, in turn, use different tools, like the Integrative Genomics Viewer (IGV), for further analysis.

\sphinxAtStartPar
This culminates the tutorial.


\section{References}
\label{\detokenize{index:references}}
\sphinxAtStartPar
\sphinxstylestrong{BMC Bioinformatics Publication}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-80}{CNV\sphinxhyphen{}seq, a new method to detect copy number variation using high\sphinxhyphen{}throughput sequencing}

\end{itemize}

\sphinxAtStartPar
\sphinxstylestrong{GitHub repositories}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{https://github.com/hliang/cnv-seq}{hliang/cnv\sphinxhyphen{}seq}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://github.com/Bioconductor/copy-number-analysis/wiki/CNV-seq}{Bioconductor/copy\sphinxhyphen{}number\sphinxhyphen{}analysis}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://github.com/pbgl/pbgl-cnvseq}{pbgl/pbgl\sphinxhyphen{}cnvseq}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://github.com/amora197/pbgl-cnvseq}{amora197/pbgl\sphinxhyphen{}cnvseq}

\end{itemize}



\renewcommand{\indexname}{Index}
\printindex
\end{document}