

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>PBGL CNV-seq Analysis &mdash; pbgl-cnvseq 1.0 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="#" class="icon icon-home"> pbgl-cnvseq
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">PBGL CNV-seq Analysis</a><ul>
<li><a class="reference internal" href="#background">Background</a></li>
<li><a class="reference internal" href="#installations-virtual-environments-and-software-packages">Installations - Virtual Environments and Software Packages</a><ul>
<li><a class="reference internal" href="#miniconda3-conda-and-mamba">Miniconda3 (conda) and Mamba</a></li>
<li><a class="reference internal" href="#git-installation-and-repo-cloning">Git Installation and Repo Cloning</a></li>
<li><a class="reference internal" href="#required-libraries-with-mamba">Required Libraries with Mamba</a><ul>
<li><a class="reference internal" href="#automatically-faster">Automatically (faster)</a></li>
<li><a class="reference internal" href="#manually-slower">Manually (slower)</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#running-jupyter">Running Jupyter</a><ul>
<li><a class="reference internal" href="#editing-the-configuration-file">Editing the Configuration File</a></li>
<li><a class="reference internal" href="#running-a-rcnv-seq-template-jupyter-notebook">Running a RCNV_seq-template Jupyter Notebook</a><ul>
<li><a class="reference internal" href="#installing-required-libraries-optional">Installing Required Libraries (optional)</a></li>
<li><a class="reference internal" href="#loading-required-libraries-mandatory">Loading Required Libraries (mandatory)</a></li>
<li><a class="reference internal" href="#user-input-mandatory">User Input (mandatory)</a></li>
<li><a class="reference internal" href="#cnv-calculations">CNV Calculations</a></li>
<li><a class="reference internal" href="#plotting">Plotting</a></li>
<li><a class="reference internal" href="#plotting-a-zoomed-in-region-of-one-chromosome">Plotting a Zoomed-In Region of One Chromosome</a></li>
</ul>
</li>
</ul>
</li>
<li><a class="reference internal" href="#example-tutorial-cnv-analysis-of-sorghum">Example Tutorial - CNV Analysis of Sorghum</a><ul>
<li><a class="reference internal" href="#data-download">Data Download</a><ul>
<li><a class="reference internal" href="#linux-terminal">Linux Terminal</a></li>
<li><a class="reference internal" href="#web-browser">Web-Browser</a></li>
</ul>
</li>
<li><a class="reference internal" href="#general-installations">General Installations</a></li>
<li><a class="reference internal" href="#github-repository-cloning">Github Repository Cloning</a></li>
<li><a class="reference internal" href="#virtual-environment-creation">Virtual Environment Creation</a></li>
<li><a class="reference internal" href="#configuration-file-editing">Configuration File Editing</a></li>
<li><a class="reference internal" href="#jupyter-notebook-analysis">Jupyter Notebook Analysis</a></li>
</ul>
</li>
<li><a class="reference internal" href="#references">References</a></li>
</ul>
</li>
</ul>
</div>
            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">pbgl-cnvseq</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="#" class="icon icon-home"></a> &raquo;</li>
        
      <li>PBGL CNV-seq Analysis</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="pbgl-cnv-seq-analysis">
<h1>PBGL CNV-seq Analysis<a class="headerlink" href="#pbgl-cnv-seq-analysis" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<div class="section" id="background">
<h2>Background<a class="headerlink" href="#background" title="Permalink to this headline">¶</a></h2>
<p>Copy number variation (CNV) analysis using CNVseq, R, Jupyter Notebooks, Miniconda3, Mamba, and Git.</p>
<p>All the commands run in a Linux terminal are preceded by the <cite>$</cite> prompt sign. To run a command, copy/past the command without the <cite>$</cite> sign. Those commands run in a Jupyter Notebook are preceded by the <cite>In [ ]:</cite></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.</p>
</div>
</div>
<div class="section" id="installations-virtual-environments-and-software-packages">
<h2>Installations - Virtual Environments and Software Packages<a class="headerlink" href="#installations-virtual-environments-and-software-packages" title="Permalink to this headline">¶</a></h2>
<p>Before installing any necessary software, it is recommended to check if the computer is running 32-bit or 64-bit for downloading Miniconda3. Run the following to verify the system:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ uname -m
</pre></div>
</div>
<div class="section" id="miniconda3-conda-and-mamba">
<h3>Miniconda3 (conda) and Mamba<a class="headerlink" href="#miniconda3-conda-and-mamba" title="Permalink to this headline">¶</a></h3>
<p>Download the Miniconda3, or simply “conda”, installer:</p>
<blockquote>
<div><ul class="simple">
<li><p><a class="reference external" href="https://docs.conda.io/en/latest/miniconda.html#linux-installers">Miniconda3 installer for Linux</a></p></li>
</ul>
</div></blockquote>
<p>Run the downloaded installer (for a 64-bit system):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bash Miniconda3-latest-Linux-x86_64.sh
</pre></div>
</div>
<p>Open a new terminal window for conda to take effect. The word <cite>(base)</cite> should appear in front of the computer name in the terminal window, like so:</p>
<div class="figure align-center">
<img alt="terminal-base" src="_images/terminal_base_env.png" />
</div>
<div class="figure align-center">
<img alt="terminal-base-detail" src="_images/terminal_base_env_detail.png" />
</div>
<p>Verify the installation and update conda in new terminal window with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ conda env list
$ conda update --all
$ conda upgrade --all
</pre></div>
</div>
<p>Install mamba library/package manager that will be used for installing software dependencies of the tool:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ conda install mamba --yes
</pre></div>
</div>
</div>
<div class="section" id="git-installation-and-repo-cloning">
<h3>Git Installation and Repo Cloning<a class="headerlink" href="#git-installation-and-repo-cloning" title="Permalink to this headline">¶</a></h3>
<p>Git is required for cloning locally (downloading a copy to your local computer) the PBGL CNVseq Github repository. Git and Github are used for version control of software. It keeps track of development, releases, and issues of a software project.</p>
<p>Install <strong>git</strong> for cloning the <strong>pbgl-cnvseq</strong> software repository from Github, where the latetest version of the tool resides:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ mamba install git --yes
</pre></div>
</div>
<p>After the instalation, clone PBGL’s CNVseq repository, <strong>pbgl-cnvseq</strong>, to the local computer in any desired directory.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ git clone https://github.com/pbgl/pbgl-cnvseq.git
</pre></div>
</div>
<p>The cloning process will depict the following:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Cloning</span> <span class="n">into</span> <span class="s1">&#39;pbgl-cnvseq&#39;</span><span class="o">...</span>
<span class="n">remote</span><span class="p">:</span> <span class="n">Enumerating</span> <span class="n">objects</span><span class="p">:</span> <span class="mi">628</span><span class="p">,</span> <span class="n">done</span><span class="o">.</span>
<span class="n">remote</span><span class="p">:</span> <span class="n">Counting</span> <span class="n">objects</span><span class="p">:</span> <span class="mi">100</span><span class="o">%</span> <span class="p">(</span><span class="mi">336</span><span class="o">/</span><span class="mi">336</span><span class="p">),</span> <span class="n">done</span><span class="o">.</span>
<span class="n">remote</span><span class="p">:</span> <span class="n">Compressing</span> <span class="n">objects</span><span class="p">:</span> <span class="mi">100</span><span class="o">%</span> <span class="p">(</span><span class="mi">239</span><span class="o">/</span><span class="mi">239</span><span class="p">),</span> <span class="n">done</span><span class="o">.</span>
<span class="n">remote</span><span class="p">:</span> <span class="n">Total</span> <span class="mi">628</span> <span class="p">(</span><span class="n">delta</span> <span class="mi">109</span><span class="p">),</span> <span class="n">reused</span> <span class="mi">292</span> <span class="p">(</span><span class="n">delta</span> <span class="mi">78</span><span class="p">),</span> <span class="n">pack</span><span class="o">-</span><span class="n">reused</span> <span class="mi">292</span>
<span class="n">Receiving</span> <span class="n">objects</span><span class="p">:</span> <span class="mi">100</span><span class="o">%</span> <span class="p">(</span><span class="mi">628</span><span class="o">/</span><span class="mi">628</span><span class="p">),</span> <span class="mf">11.79</span> <span class="n">MiB</span> <span class="o">|</span> <span class="mf">7.03</span> <span class="n">MiB</span><span class="o">/</span><span class="n">s</span><span class="p">,</span> <span class="n">done</span><span class="o">.</span>
<span class="n">Resolving</span> <span class="n">deltas</span><span class="p">:</span> <span class="mi">100</span><span class="o">%</span> <span class="p">(</span><span class="mi">190</span><span class="o">/</span><span class="mi">190</span><span class="p">),</span> <span class="n">done</span><span class="o">.</span>
</pre></div>
</div>
<p>The <strong>pbgl-cnvseq</strong> repository should have been clones successfully. Verify that the download is complete by listing the folders/files in the directory.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ ls -l
</pre></div>
</div>
<p>The folder called <strong>pbgl-cnvseq</strong> should be listed in the directory.</p>
</div>
<div class="section" id="required-libraries-with-mamba">
<h3>Required Libraries with Mamba<a class="headerlink" href="#required-libraries-with-mamba" title="Permalink to this headline">¶</a></h3>
<p>CNVseq has multiple dependencies, listed below:</p>
<ul class="simple">
<li><p>Samtools</p></li>
<li><p>R</p>
<ul>
<li><p>configr</p></li>
<li><p>ggplot2</p></li>
<li><p>BiocManager</p></li>
<li><p>Bioconductor-GenomicAlignments</p></li>
<li><p>Bioconductor-GenomeInfoDb</p></li>
</ul>
</li>
<li><p>Jupyter Notebook</p>
<ul>
<li><p>IRkernel</p></li>
</ul>
</li>
</ul>
<p>The necessary R packages are installed through the Jupyter Notebook. It proves as a faster and error-free way to install <strong>configr</strong>, <strong>ggplot2</strong>, <strong>Biocmanager</strong>, <strong>Bioconductor-GenomicAlignments</strong>, and <strong>Bioconductor-GenomeInfoDb</strong> packages.</p>
<p>There are two ways to install the rest of the necessary libraries to run CNV-seq: automatically or manually.</p>
<div class="section" id="automatically-faster">
<h4>Automatically (faster)<a class="headerlink" href="#automatically-faster" title="Permalink to this headline">¶</a></h4>
<p>One YAML file, <strong>environment.yml</strong>, is provided to automatically create a virtual environment and install the dependent libraries through mamba. The file creates the <strong>cnvseq</strong> virtual environment, along R, Jupyter Notebook, and the R-kernel in Jupyter. It also installs the dependent R libraries. Run <strong>environment.yml</strong>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ mamba env create --file envs/environment.yml
</pre></div>
</div>
<p>Once done, a list of the virtual environments available can be seen by running:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ conda env list
</pre></div>
</div>
<p>Activate (enter) the recently-created virtual environment <strong>cnvseq</strong>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ conda activate cnvseq
</pre></div>
</div>
<p>Once done, the virtual environment should be activated and all the necessary packages should be installed. This can be verified with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ conda list
</pre></div>
</div>
</div>
<div class="section" id="manually-slower">
<h4>Manually (slower)<a class="headerlink" href="#manually-slower" title="Permalink to this headline">¶</a></h4>
<p>To manually create and activate an environment, run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ conda create --name cnvseq
$ conda activate cnvseq
</pre></div>
</div>
<p>Start running the installations of the necessary libraries:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ mamba install --channel conda-forge notebook r-irkernel r-biocmanager --yes
$ mamba install --channel bioconda samtools bioconductor-genomicalignments bioconductor-genomeinfodb --yes
$ mamba install --channel r r-ggplot2 --yes
$ mamba install --channel pcgr r-configr --yes
</pre></div>
</div>
<p>Once done, all the necessary packages should be installed. This can be verified with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ conda list
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="running-jupyter">
<h2>Running Jupyter<a class="headerlink" href="#running-jupyter" title="Permalink to this headline">¶</a></h2>
<p>To activate Jupyter, run the following in the terminal:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ jupyter notebook
</pre></div>
</div>
<p>This command will start a Jupyter session inside the directory the command is run. The user can navigate between directories, visualize files, and edit files in a web browser by clicking on directories or files, respectively.</p>
<p>Look for the directory <strong>pbgl-cnvseq</strong> and click on it. Click on <strong>tool</strong> directory, which contains three directories and two Jupyter Notebooks. Here is a breakdown of each:</p>
<ul class="simple">
<li><p><cite>config</cite>:</p>
<ul>
<li><p>directory containing configuration files specifying file paths, parameter definitions, and comparison lists</p></li>
</ul>
</li>
<li><p><cite>helper-functions</cite>:</p>
<ul>
<li><p>directory containing R scripts with functions to calculate and plot CNVs</p></li>
</ul>
</li>
<li><p><cite>output</cite>:</p>
<ul>
<li><p>directory that will contain both tab-files and images output after running a CNVseq analysis</p></li>
</ul>
</li>
<li><p>two Jupyter Notebooks:</p>
<ul>
<li><p>RCNV-seq-PBGL-sorghum-analysis-example.ipynb</p>
<ul>
<li><p>example analysis of a comparison between a control and mutant of sorghum</p></li>
</ul>
</li>
<li><p>RCNV-seq-template.ipynb</p>
<ul>
<li><p>template for the user</p></li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Jupyter lets the user duplicate, rename, move, download, view, or edit files in a web browser. This can be done by clicking the box next to a file and choosing accordingly.</p>
</div>
<div class="section" id="editing-the-configuration-file">
<h3>Editing the Configuration File<a class="headerlink" href="#editing-the-configuration-file" title="Permalink to this headline">¶</a></h3>
<p>In order to run the CNVseq Jupyter Notebook, the user needs to feed it with a configuration file (<strong>config-CNVseq.yml</strong>) that specifies the paths to the bam files, comparisons to be done, chromosomes to analyze, and parameter definitions for calculating and plotting CNVs.</p>
<p>The configuration file <strong>config-CNVseq.yml</strong> can be found in the <strong>pbgl-cnvseq/tool/config</strong> directory. The configuration file contains the following:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># parameters for CNV/window size calculations defaults from HLiang:</span>
<span class="c1"># https://github.com/hliang/cnv-seq</span>
<span class="n">parameters</span><span class="p">:</span>
  <span class="n">annotate</span><span class="p">:</span> <span class="n">TRUE</span>
  <span class="n">bed_file_present</span><span class="p">:</span> <span class="n">FALSE</span>
  <span class="n">bigger</span><span class="p">:</span> <span class="mf">1.5</span>
  <span class="n">log2</span><span class="p">:</span> <span class="mf">0.6</span>
  <span class="n">pvalue</span><span class="p">:</span> <span class="mf">0.001</span>
  <span class="n">window_size</span><span class="p">:</span> <span class="mi">10000</span>

<span class="c1"># list of chromosomes to be analyzed</span>
<span class="n">chromosomes</span><span class="p">:</span>
  <span class="o">-</span> <span class="n">Chromosome1</span>
  <span class="o">-</span> <span class="n">Chromosome2</span>
  <span class="o">-</span> <span class="n">Chromosome3</span>
  <span class="o">-</span> <span class="n">AnotherChromosome</span>

<span class="c1"># folder to store outputs; should be inside &quot;output&quot; directory, example:</span>
<span class="c1"># output_path: output/run-name or output/organism</span>
<span class="n">output_path</span><span class="p">:</span> <span class="n">output</span><span class="o">/</span><span class="n">organism</span><span class="o">-</span><span class="n">being</span><span class="o">-</span><span class="n">analyzed</span>

<span class="c1"># path to bed file for varying window sizes (optional)</span>
<span class="n">bed_path</span><span class="p">:</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">bed</span><span class="o">/</span><span class="n">file</span><span class="o">.</span><span class="n">bed</span>

<span class="c1"># paths to bam files</span>
<span class="n">paths</span><span class="p">:</span>
  <span class="n">control</span><span class="o">-</span><span class="mi">1</span><span class="p">:</span> <span class="o">&amp;</span><span class="n">control</span><span class="o">-</span><span class="mi">1</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">bam</span><span class="o">/</span><span class="n">file</span><span class="o">/</span><span class="n">control</span><span class="o">-</span><span class="mf">1.</span><span class="n">bam</span>
  <span class="n">mutant</span><span class="o">-</span><span class="mi">1</span><span class="p">:</span> <span class="o">&amp;</span><span class="n">mutant</span><span class="o">-</span><span class="mi">1</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">bam</span><span class="o">/</span><span class="n">file</span><span class="o">/</span><span class="n">mutant</span><span class="o">-</span><span class="mf">1.</span><span class="n">bam</span>
  <span class="n">mutant</span><span class="o">-</span><span class="mi">2</span><span class="p">:</span> <span class="o">&amp;</span><span class="n">mutant</span><span class="o">-</span><span class="mi">2</span> <span class="o">/</span><span class="n">path</span><span class="o">/</span><span class="n">to</span><span class="o">/</span><span class="n">bam</span><span class="o">/</span><span class="n">file</span><span class="o">/</span><span class="n">mutant</span><span class="o">-</span><span class="mf">2.</span><span class="n">bam</span>

<span class="c1"># comparisons to be analyzed</span>
<span class="n">comparisons</span><span class="p">:</span>
  <span class="n">control</span><span class="o">-</span><span class="mi">1</span><span class="o">-</span><span class="n">vs</span><span class="o">-</span><span class="n">mutant</span><span class="o">-</span><span class="mi">1</span><span class="p">:</span>
    <span class="n">control</span><span class="p">:</span> <span class="o">*</span><span class="n">control</span><span class="o">-</span><span class="mi">1</span>
    <span class="n">mutant</span><span class="p">:</span> <span class="o">*</span><span class="n">mutant</span><span class="o">-</span><span class="mi">1</span>
  <span class="n">control</span><span class="o">-</span><span class="mi">1</span><span class="o">-</span><span class="n">vs</span><span class="o">-</span><span class="n">mutant</span><span class="o">-</span><span class="mi">2</span><span class="p">:</span>
    <span class="n">control</span><span class="p">:</span> <span class="o">*</span><span class="n">control</span><span class="o">-</span><span class="mi">1</span>
    <span class="n">mutant</span><span class="p">:</span> <span class="o">*</span><span class="n">mutant</span><span class="o">-</span><span class="mi">2</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The user needs to edit <strong>config-CNVseq.yml</strong> to point towards bam/bed files; specify comparisons and chromosomes to analyze; and define the parameters to calculate/plot CNVs.</p>
</div>
<p>One example configuration files is provided (<strong>config-CNVseq-PBGL-sorghum-analysis-example.yml</strong>). The configuration file <strong>config-CNVseq.yml</strong> contains multiple fields to be defined by the user.</p>
<ul class="simple">
<li><p><cite>parameters</cite>:</p>
<ul>
<li><p>parameters used to create window sizes, thresholds, plots, etc</p></li>
<li><p>the parameter defaults are provided accroding to HLiang’s original values</p></li>
</ul>
</li>
<li><p><cite>chromosomes</cite>:</p>
<ul>
<li><p>list of chromosome names to analyze</p></li>
<li><p>chromosome names can be found in a bam file’s header using the following samtools command:</p></li>
</ul>
</li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools view -h my_bam_file.bam | less -S
</pre></div>
</div>
<ul class="simple">
<li><p><cite>output_path</cite>:</p>
<ul>
<li><p>directory that will contain both tab-files and images output after running a CNVseq analysis</p></li>
<li><p>the defined path will be inside the <strong>pbgl-cnvseq/tool/output</strong> directory</p></li>
<li><p><cite>images</cite>:</p>
<ul>
<li><p>directory that will store the CNV image outputs per comparison of:</p></li>
<li><p>all chromosomes in one plot</p></li>
<li><p>each chromosome individually in one plot</p></li>
<li><p>any zoomed-in region plot of one chromosome</p></li>
</ul>
</li>
<li><p><cite>tab-files</cite>:</p>
<ul>
<li><p>directory that will contain two types of output tab-delimited files:</p>
<ul>
<li><p>hits used to calculate CNVs, containing chromosome, start, end, width, control coverage, mutant coverage</p></li>
<li><p>CNVs per chromosome per comparison, containing CNV number, chromosome, start, end, size, log2, and p-value</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li><p><cite>bed_path</cite>:</p>
<ul>
<li><p>path pointing to bed file containing targets if using varying window sizes</p></li>
<li><p>it is recommended to run the analysis without a <strong>.bed</strong> file; plots will only reflect the targets in a <strong>.bed</strong> file</p></li>
<li><p>if a <strong>.bed</strong> file is provided, the <cite>bed_file_present</cite> parameter under the <cite>parameters</cite> section has to be changed to <cite>TRUE</cite></p></li>
</ul>
</li>
<li><p><cite>paths</cite>:</p>
<ul>
<li><p>sample names and their respective paths to <strong>.bam</strong> files</p></li>
<li><p>samples can be named as desired but the sample name must be repeated after the colon and prefixed with a <cite>&amp;</cite> sign</p></li>
<li><p>the <cite>&amp;</cite> prefix sign is used to reference the sample’s path in different places of the same configuration file</p></li>
<li><p>example use:</p></li>
</ul>
</li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">paths</span><span class="p">:</span>
  <span class="n">mysample</span><span class="p">:</span> <span class="o">&amp;</span><span class="n">mysample</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">username</span><span class="o">/</span><span class="n">bam_files</span><span class="o">/</span><span class="n">mysample</span><span class="o">.</span><span class="n">bam</span>
  <span class="n">XYZ</span><span class="o">-</span><span class="mi">123</span><span class="p">:</span> <span class="o">&amp;</span><span class="n">XYZ</span><span class="o">-</span><span class="mi">123</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">username</span><span class="o">/</span><span class="n">bam_files</span><span class="o">/</span><span class="n">XYZ</span><span class="o">-</span><span class="mf">123.</span><span class="n">bam</span>
  <span class="n">potato95</span><span class="p">:</span> <span class="o">&amp;</span><span class="n">potato95</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">username</span><span class="o">/</span><span class="n">bam_files</span><span class="o">/</span><span class="n">potato95</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<ul class="simple">
<li><p><cite>comparisons</cite>:</p>
<ul>
<li><p>comparison names with respective control and mutant samples per comparison</p></li>
<li><p>each comparison can be named as desired</p></li>
<li><p>the sample names to be used as <cite>control</cite> and <cite>mutant</cite> need to be prefixed by a <cite>*</cite> sign</p></li>
<li><p>the <cite>*</cite> prefixed sign is used to extract the sample’s path defined in the <cite>paths</cite> section</p></li>
<li><p>example:</p></li>
</ul>
</li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">comparisons</span><span class="p">:</span>
  <span class="n">comparison</span><span class="o">-</span><span class="mi">1</span><span class="p">:</span>
    <span class="n">control</span><span class="p">:</span> <span class="o">*</span><span class="n">mysample</span>
    <span class="n">mutant</span><span class="p">:</span> <span class="o">*</span><span class="n">potato95</span>
  <span class="n">a</span><span class="o">-</span><span class="n">different</span><span class="o">-</span><span class="n">comparison</span><span class="o">-</span><span class="mi">278</span><span class="n">asd</span><span class="p">:</span>
    <span class="n">control</span><span class="p">:</span> <span class="o">*</span><span class="n">mysample</span>
    <span class="n">mutant</span><span class="p">:</span> <span class="o">*</span><span class="n">XYZ</span><span class="o">-</span><span class="mi">123</span>
</pre></div>
</div>
</div>
<div class="section" id="running-a-rcnv-seq-template-jupyter-notebook">
<h3>Running a RCNV_seq-template Jupyter Notebook<a class="headerlink" href="#running-a-rcnv-seq-template-jupyter-notebook" title="Permalink to this headline">¶</a></h3>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>It is recommended to duplicate the <strong>RCNV-seq-template</strong> notebook and then renaming the copy before doing any edits to the notebook.</p>
</div>
<p>In the <strong>pbgl-cvnseq/tool</strong> directory, click on <strong>RCNV-seq-template</strong> and a new tab in your web-browser will open the notebook.</p>
<p>The notebook contains cells that are populated by text or code. Instructions are provided in the notebook to guide the user. To run a cell, click on the corresponding cell and click on the <cite>Run</cite> button on the top of the notebook. Another way to run a cell can be done by clicking on the corresponding cell and pressing <strong>Ctrl + Enter</strong> or <strong>Shift + Enter</strong>.</p>
<p>The notebook consists of 6 sections:</p>
<ol class="arabic simple">
<li><p>Installing Required Libraries (optional)</p></li>
<li><p>Loading Required Libraries (mandatory)</p></li>
<li><p>User Input (mandatory)</p></li>
<li><p>CNV Calculations</p></li>
<li><p>Plotting</p></li>
<li><p>Plotting a Zoomed-In Region of One Chromosome</p></li>
</ol>
<div class="section" id="installing-required-libraries-optional">
<h4>Installing Required Libraries (optional)<a class="headerlink" href="#installing-required-libraries-optional" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li><p>libraries being installed:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>In [ ]: # install necessary libraries using R functions
        if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
            install.packages(&quot;BiocManager&quot;)

        BiocManager::install(c(&quot;GenomicAlignments&quot;, &quot;GenomeInfoDb&quot;))
        install.packages(&quot;configr&quot;)
        install.packages(&quot;ggplot2&quot;)
</pre></div>
</div>
<ul class="simple">
<li><p>to be run if the mamba installations were not successful or the loading of the required libraries fails under the <strong>Loading Required Libraries</strong> section</p></li>
</ul>
</div>
<div class="section" id="loading-required-libraries-mandatory">
<h4>Loading Required Libraries (mandatory)<a class="headerlink" href="#loading-required-libraries-mandatory" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li><p>libraries being loaded:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="c1"># load necessary libraries</span>
        <span class="n">library</span><span class="p">(</span><span class="n">GenomicAlignments</span><span class="p">)</span>
        <span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
        <span class="n">library</span><span class="p">(</span><span class="n">configr</span><span class="p">)</span>

        <span class="c1"># specify source R script with helper functions</span>
        <span class="n">source</span><span class="p">(</span><span class="s2">&quot;helper-functions/RCNV_seq-helper.R&quot;</span><span class="p">)</span>
        <span class="n">source</span><span class="p">(</span><span class="s2">&quot;helper-functions/cnvHLiang.R&quot;</span><span class="p">)</span>
</pre></div>
</div>
<ul class="simple">
<li><p>this cell will load necessary libraries and scripts containing the necessary functions to be used</p></li>
<li><p>if running this cell fails, some libraries may be missing from the installation</p>
<ul>
<li><p>to fix this issue, run the installations under the <strong>Installing Required Libraries</strong></p></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="user-input-mandatory">
<h4>User Input (mandatory)<a class="headerlink" href="#user-input-mandatory" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li><p>the user needs to write the appropriate name of the configuration file being used in the following cell:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">configPath</span> <span class="o">&lt;-</span> <span class="s2">&quot;config/config-CNVseq.yml&quot;</span>
</pre></div>
</div>
<ul class="simple">
<li><p>the bottom cell will extract the fields defined in the configuration file:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">config</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">config</span><span class="p">(</span><span class="n">configPath</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="cnv-calculations">
<h4>CNV Calculations<a class="headerlink" href="#cnv-calculations" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li><p>function to calculate all the hits and CNVs:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">cnvCalculate</span><span class="p">(</span><span class="n">config</span><span class="p">)</span>
</pre></div>
</div>
<ul class="simple">
<li><p>output tabulated files are stored inside <strong>pbgl-cnvseq/tool/output/name_defined_in_output_path_of_config/tab-files</strong> directory</p></li>
</ul>
</div>
<div class="section" id="plotting">
<h4>Plotting<a class="headerlink" href="#plotting" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li><p>function to plot two types of images:</p>
<ol class="arabic simple">
<li><p>CNVs of all chromosomes in the same plot</p></li>
<li><p>CNVs of one chromosome per plot</p></li>
</ol>
</li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">cnvPlot</span><span class="p">(</span><span class="n">config</span><span class="p">,</span> <span class="n">imgType</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">yMin</span><span class="o">=</span> <span class="p">,</span> <span class="n">yMax</span><span class="o">=</span> <span class="p">)</span>
</pre></div>
</div>
<ul class="simple">
<li><p>function parameters are:</p>
<ul>
<li><p><cite>config</cite> - configuration file defined under <strong>User Input</strong> section</p></li>
<li><p><cite>imgType</cite> -  image extention to use; available options are: <cite>png</cite>, <cite>jpeg</cite>, <cite>svg</cite>, and <cite>pdf</cite>; default to <cite>png</cite></p></li>
<li><p><cite>yMin</cite> - y-axis bottom limit; default to <cite>-5</cite></p></li>
<li><p><cite>yMax</cite> - y-axis upper limit; default to <cite>5</cite></p></li>
</ul>
</li>
<li><p>output images are stored inside the <strong>pbgl-cnvseq/tool/output/name_defined_in_output_path_of_config/images</strong> directory</p></li>
<li><p>it is recommended to inspect the CNV-plots with default y-limits and then modify</p></li>
<li><p><cite>cnvPlotting</cite> can be used in two ways: with default values or user-defined values</p>
<ul>
<li><p>with default values, the parameters can be omitted and set to <cite>imgType=”png”</cite>, <cite>yMin=-5</cite> and <cite>yMax=5</cite></p></li>
<li><p>the following two commands have the same parameter values and will output the same plots:</p></li>
</ul>
</li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">cnvPlot</span><span class="p">(</span><span class="n">config</span><span class="p">)</span>
        <span class="n">cnvPlot</span><span class="p">(</span><span class="n">config</span><span class="p">,</span> <span class="n">imgType</span><span class="o">=</span><span class="s2">&quot;png&quot;</span><span class="p">,</span> <span class="n">yMin</span><span class="o">=-</span><span class="mi">5</span><span class="p">,</span> <span class="n">yMax</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="plotting-a-zoomed-in-region-of-one-chromosome">
<h4>Plotting a Zoomed-In Region of One Chromosome<a class="headerlink" href="#plotting-a-zoomed-in-region-of-one-chromosome" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li><p>function to plot a specific zoomed-in region of one chromosome</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">cnvPlotZoom</span><span class="p">(</span><span class="n">config</span><span class="p">,</span> <span class="n">tabFile</span><span class="p">,</span> <span class="n">chromosome</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">,</span> <span class="n">start</span><span class="o">=</span> <span class="p">,</span> <span class="n">end</span><span class="o">=</span> <span class="p">,</span> <span class="n">yMin</span><span class="o">=</span> <span class="p">,</span> <span class="n">yMax</span><span class="o">=</span> <span class="p">,</span> <span class="n">imgType</span><span class="o">=</span><span class="s2">&quot;&quot;</span><span class="p">)</span>
</pre></div>
</div>
<ul class="simple">
<li><p>function parameters are:</p>
<ul>
<li><p><cite>config</cite> - configuration file defined under <strong>User Input</strong> section</p></li>
<li><p><cite>tabFile</cite> -  tabulated file containing all hits; requires user-input to define path to all-hits tabulated file</p></li>
<li><p><cite>chromosome</cite> - chromosome name to focus on; default to <cite>NA</cite></p></li>
<li><p><cite>start</cite> - start of window in bp; both scientific notation is accepted: <cite>100000</cite> or <cite>10e4</cite>; defaults to <cite>NA</cite></p></li>
<li><p><cite>end</cite> - end of window in bp; the same applies as in the <cite>start</cite> parameter; defaults to NA</p></li>
<li><p><cite>yMin</cite> - y-axis bottom limit; default to <cite>-5</cite></p></li>
<li><p><cite>yMax</cite> - y-axis upper limit; default to <cite>5</cite></p></li>
<li><p><cite>imgType</cite> -  image extention to use; available options are: <cite>png</cite>, <cite>jpeg</cite>, <cite>svg</cite>, and <cite>pdf</cite>; defaults to <cite>png</cite></p></li>
</ul>
</li>
<li><p>requires path definition of <cite>tabFile</cite> parameter in the cell:</p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">tabFile</span> <span class="o">&lt;-</span> <span class="s2">&quot;output/run-name/tab-files/tab-file.tab&quot;</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="example-tutorial-cnv-analysis-of-sorghum">
<h2>Example Tutorial - CNV Analysis of Sorghum<a class="headerlink" href="#example-tutorial-cnv-analysis-of-sorghum" title="Permalink to this headline">¶</a></h2>
<p>In this section of the manual, an example analysis of sorghum will be shown in a step-by-step process. The data has been subset to three chromosomes Chr04, Chr05, and Chr09. The CNVseq analysis will depict a deletion on chromosome Chr09.</p>
<p>The tutorial is divided between the following sections:</p>
<ol class="arabic simple">
<li><p>Data Download</p></li>
<li><p>General Installations</p></li>
<li><p>Github Repository Cloning</p></li>
<li><p>Virtual Environment Creation</p></li>
<li><p>Configuration File Editing</p></li>
<li><p>Jupyter Notebook Analysis</p></li>
</ol>
<div class="section" id="data-download">
<h3>Data Download<a class="headerlink" href="#data-download" title="Permalink to this headline">¶</a></h3>
<p>Running this tutorial requires two binary alignment map (BAM) files of sorghum crop: one control and one mutant. Each <strong>.bam</strong> file is subset for chromosomes Chr04, Chr05, and Chr09; this is reflected on their names. They have been both sorted and indexed using samtools.</p>
<p>The following links can be used to download the necessary <strong>.bam</strong> and <strong>.bam.bai</strong> (index) files:</p>
<ul class="simple">
<li><p>con-2_S1 (sorghum control)</p>
<ul>
<li><p><a class="reference external" href="https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/con-2_S1-Chromes-04-05-09.bam">https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/con-2_S1-Chromes-04-05-09.bam</a></p></li>
<li><p><a class="reference external" href="https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/con-2_S1-Chromes-04-05-09.bam.bai">https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/con-2_S1-Chromes-04-05-09.bam.bai</a></p></li>
</ul>
</li>
<li><p>D2-1_S7 (sorghum mutant)</p>
<ul>
<li><p><a class="reference external" href="https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/D2-1_S7-Chromes-04-05-09.bam">https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/D2-1_S7-Chromes-04-05-09.bam</a></p></li>
<li><p><a class="reference external" href="https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/D2-1_S7-Chromes-04-05-09.bam.bai">https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/D2-1_S7-Chromes-04-05-09.bam.bai</a></p></li>
</ul>
</li>
</ul>
<p>There are two additional ways to download the <strong>.bam</strong> files and their respectives indices, besides clicking the links above:</p>
<ol class="arabic simple">
<li><p>Linux terminal</p></li>
<li><p>Web-Browser</p></li>
</ol>
<div class="section" id="linux-terminal">
<h4>Linux Terminal<a class="headerlink" href="#linux-terminal" title="Permalink to this headline">¶</a></h4>
<p>Open a new terminal and navigate to a directory of choice. We recommend creating a directory to store the data and running <cite>wget</cite> in the respective location:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ mkdir bam_files
$ cd bam_files
$ wget https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/con-2_S1-Chromes-04-05-09.bam
$ wget https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/con-2_S1-Chromes-04-05-09.bam.bai
$ wget https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/D2-1_S7-Chromes-04-05-09.bam
$ wget https://bss1innov1nafa1poc1.blob.core.windows.net/sample-container/2021_Training/D2-1_S7-Chromes-04-05-09.bam.bai
</pre></div>
</div>
</div>
<div class="section" id="web-browser">
<h4>Web-Browser<a class="headerlink" href="#web-browser" title="Permalink to this headline">¶</a></h4>
<p>Open a web-browser of preference. Copy/paste the links provided above in the address bar. This should automatically begin the download. Move the downloaded files to a location of personal preference.</p>
</div>
</div>
<div class="section" id="general-installations">
<h3>General Installations<a class="headerlink" href="#general-installations" title="Permalink to this headline">¶</a></h3>
<p>Open a web browser and copy/paste the following link to download Miniconda3:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://docs.conda.io/en/latest/miniconda.html#linux-installers">https://docs.conda.io/en/latest/miniconda.html#linux-installers</a></p></li>
</ul>
<p>After download, open a new terminal window, navigate to the directory with the downloaded Miniconda3 installer, and run the installation.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The donwloaded Miniconda3 installer file might not match the one run in this example. Please, type the corresponding name of the <strong>.sh</strong> file downloaded.</p>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bash Miniconda3-latest-Linux-x86_64.sh
</pre></div>
</div>
<p>Once the Miniconda3 installation is done, close the terminal and open a new one. The word <cite>(base)</cite> should be present to the left of the computer name in the prompt. Update conda and install mamba.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ conda update --all --yes
$ conda upgrade --all --yes
$ conda install mamba --yes
</pre></div>
</div>
</div>
<div class="section" id="github-repository-cloning">
<h3>Github Repository Cloning<a class="headerlink" href="#github-repository-cloning" title="Permalink to this headline">¶</a></h3>
<p>Git or Github is used for storage and version control of software projects. Git is used to manage in a local Linux terminal. First, git will need to be installed. In a new terminal window, run the installation command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ mamba install git --yes
</pre></div>
</div>
<p>After installing git, create a location to store Github repoitories, navigate into it, and clone (download a copy locally) of PBGL’s <strong>pbgl-cnvseq</strong> repository.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ mkdir Github
$ cd Github
$ git clone https://github.com/pbgl/pbgl-cnvseq
</pre></div>
</div>
<p>A successful cloning will output the following:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Cloning</span> <span class="n">into</span> <span class="s1">&#39;pbgl-cnvseq&#39;</span><span class="o">...</span>
<span class="n">remote</span><span class="p">:</span> <span class="n">Enumerating</span> <span class="n">objects</span><span class="p">:</span> <span class="mi">628</span><span class="p">,</span> <span class="n">done</span><span class="o">.</span>
<span class="n">remote</span><span class="p">:</span> <span class="n">Counting</span> <span class="n">objects</span><span class="p">:</span> <span class="mi">100</span><span class="o">%</span> <span class="p">(</span><span class="mi">336</span><span class="o">/</span><span class="mi">336</span><span class="p">),</span> <span class="n">done</span><span class="o">.</span>
<span class="n">remote</span><span class="p">:</span> <span class="n">Compressing</span> <span class="n">objects</span><span class="p">:</span> <span class="mi">100</span><span class="o">%</span> <span class="p">(</span><span class="mi">239</span><span class="o">/</span><span class="mi">239</span><span class="p">),</span> <span class="n">done</span><span class="o">.</span>
<span class="n">remote</span><span class="p">:</span> <span class="n">Total</span> <span class="mi">628</span> <span class="p">(</span><span class="n">delta</span> <span class="mi">109</span><span class="p">),</span> <span class="n">reused</span> <span class="mi">292</span> <span class="p">(</span><span class="n">delta</span> <span class="mi">78</span><span class="p">),</span> <span class="n">pack</span><span class="o">-</span><span class="n">reused</span> <span class="mi">292</span>
<span class="n">Receiving</span> <span class="n">objects</span><span class="p">:</span> <span class="mi">100</span><span class="o">%</span> <span class="p">(</span><span class="mi">628</span><span class="o">/</span><span class="mi">628</span><span class="p">),</span> <span class="mf">11.79</span> <span class="n">MiB</span> <span class="o">|</span> <span class="mf">7.03</span> <span class="n">MiB</span><span class="o">/</span><span class="n">s</span><span class="p">,</span> <span class="n">done</span><span class="o">.</span>
<span class="n">Resolving</span> <span class="n">deltas</span><span class="p">:</span> <span class="mi">100</span><span class="o">%</span> <span class="p">(</span><span class="mi">190</span><span class="o">/</span><span class="mi">190</span><span class="p">),</span> <span class="n">done</span><span class="o">.</span>
</pre></div>
</div>
<p>Navigate into the cloned repository,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ cd pbgl-cnvseq
</pre></div>
</div>
</div>
<div class="section" id="virtual-environment-creation">
<h3>Virtual Environment Creation<a class="headerlink" href="#virtual-environment-creation" title="Permalink to this headline">¶</a></h3>
<p>Once inside the <strong>pbgl-cnvseq</strong> directory, create the <strong>cnvseq</strong> virtual environment, install the necessary libraries, and activate the newly created <strong>cnvseq</strong> virtual environment.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ mamba env create --file envs/environment.yml
$ conda activate cnvseq
</pre></div>
</div>
<p>The name <cite>(cnvseq)</cite> environment should be reflected to the left of the computer name in the terminal command prompt.</p>
</div>
<div class="section" id="configuration-file-editing">
<h3>Configuration File Editing<a class="headerlink" href="#configuration-file-editing" title="Permalink to this headline">¶</a></h3>
<p>Open a Jupyter session by running,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ jupyter notebook
</pre></div>
</div>
<p>Navigate to <strong>pbgl-cnvseq/tool/config</strong> and click on <strong>config-CNVseq-PBGL-sorghum-analysis-example.yml</strong>. This will open the configuration file in a new web-browser tab. Copy/paste the following in the configuration file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The paths to the bam files will not match. They need to be edited accordingly to point towards the location of the bam files stored locally in the user’s computer.</p>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># parameters for CNV/window size calculations defaults from HLiang:</span>
<span class="c1"># https://github.com/hliang/cnv-seq</span>
<span class="n">parameters</span><span class="p">:</span>
  <span class="n">annotate</span><span class="p">:</span> <span class="n">TRUE</span>
  <span class="n">bed_file_present</span><span class="p">:</span> <span class="n">FALSE</span>
  <span class="n">bigger</span><span class="p">:</span> <span class="mf">1.5</span>
  <span class="n">log2</span><span class="p">:</span> <span class="mf">0.6</span>
  <span class="n">pvalue</span><span class="p">:</span> <span class="mf">0.001</span>
  <span class="n">window_size</span><span class="p">:</span> <span class="mi">10000</span>

<span class="c1"># list of chromosomes to be analyzed</span>
<span class="n">chromosomes</span><span class="p">:</span>
  <span class="o">-</span> <span class="n">Chr04</span>
  <span class="o">-</span> <span class="n">Chr05</span>
  <span class="o">-</span> <span class="n">Chr09</span>

<span class="c1"># folder to store outputs; should be inside &quot;output&quot; directory, example:</span>
<span class="c1"># output_path: output/run-name or output/organism</span>
<span class="n">output_path</span><span class="p">:</span> <span class="n">output</span><span class="o">/</span><span class="n">PBGL</span><span class="o">-</span><span class="n">sorghum</span><span class="o">-</span><span class="n">analysis</span><span class="o">-</span><span class="n">example</span>

<span class="c1"># path to bed file for varying window sizes (optional)</span>
<span class="n">bed_path</span><span class="p">:</span>

<span class="c1"># paths to bam files</span>
<span class="n">paths</span><span class="p">:</span>
  <span class="n">con</span><span class="o">-</span><span class="mi">2</span><span class="n">_S1</span><span class="p">:</span> <span class="o">&amp;</span><span class="n">con</span><span class="o">-</span><span class="mi">2</span><span class="n">_S1</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">anibal</span><span class="o">/</span><span class="n">bam_files</span><span class="o">/</span><span class="n">sorghum</span><span class="o">/</span><span class="n">con</span><span class="o">-</span><span class="mi">2</span><span class="n">_S1</span><span class="o">-</span><span class="n">Chromes</span><span class="o">-</span><span class="mi">04</span><span class="o">-</span><span class="mi">05</span><span class="o">-</span><span class="mf">09.</span><span class="n">bam</span>
  <span class="n">D2</span><span class="o">-</span><span class="mi">1</span><span class="n">_S7</span><span class="p">:</span> <span class="o">&amp;</span><span class="n">D2</span><span class="o">-</span><span class="mi">1</span><span class="n">_S7</span> <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">anibal</span><span class="o">/</span><span class="n">bam_files</span><span class="o">/</span><span class="n">sorghum</span><span class="o">/</span><span class="n">D2</span><span class="o">-</span><span class="mi">1</span><span class="n">_S7</span><span class="o">-</span><span class="n">Chromes</span><span class="o">-</span><span class="mi">04</span><span class="o">-</span><span class="mi">05</span><span class="o">-</span><span class="mf">09.</span><span class="n">bam</span>

<span class="c1"># comparisons to be analyzed</span>
<span class="n">comparisons</span><span class="p">:</span>
  <span class="n">con</span><span class="o">-</span><span class="mi">2</span><span class="n">_s1</span><span class="o">-</span><span class="n">vs</span><span class="o">-</span><span class="n">D2</span><span class="o">-</span><span class="mi">1</span><span class="n">_S7</span><span class="p">:</span>
    <span class="n">control</span><span class="p">:</span> <span class="o">*</span><span class="n">con</span><span class="o">-</span><span class="mi">2</span><span class="n">_S1</span>
    <span class="n">mutant</span><span class="p">:</span> <span class="o">*</span><span class="n">D2</span><span class="o">-</span><span class="mi">1</span><span class="n">_S7</span>
</pre></div>
</div>
<p>Save the file and close the tab.</p>
</div>
<div class="section" id="jupyter-notebook-analysis">
<h3>Jupyter Notebook Analysis<a class="headerlink" href="#jupyter-notebook-analysis" title="Permalink to this headline">¶</a></h3>
<p>In the open Jupyter session, navigate to the <strong>pbgl-cnvseq/tool</strong> directory and click on the <strong>RCNV-seq-template.ipynb</strong> Jupyter Notebook. Begin by running the <strong>Loading Required Libraries</strong> section. This can be done by clicking on the cell to be run followed by clicking the <cite>Run</cite> button on the top of the Jupyter Notebook.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span>  <span class="c1"># load necessary libraries</span>
         <span class="n">library</span><span class="p">(</span><span class="n">GenomicAlignments</span><span class="p">)</span>
         <span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
         <span class="n">library</span><span class="p">(</span><span class="n">configr</span><span class="p">)</span>

         <span class="c1"># specify source R script with helper functions</span>
         <span class="n">source</span><span class="p">(</span><span class="s2">&quot;helper-functions/RCNV_seq-helper.R&quot;</span><span class="p">)</span>
         <span class="n">source</span><span class="p">(</span><span class="s2">&quot;helper-functions/cnvHLiang.R&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>If loading the libraries is unsuccessful (specified in the cell output), an attempt to install the missing libraries can be done by running the <strong>Installing Required Libraries</strong> section,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>In [ ]:  # install necessary libraries using R functions
         if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
             install.packages(&quot;BiocManager&quot;)

         BiocManager::install(c(&quot;GenomicAlignments&quot;, &quot;GenomeInfoDb&quot;))
         install.packages(&quot;configr&quot;)
         install.packages(&quot;ggplot2&quot;)
</pre></div>
</div>
<p>Run the <strong>User Input</strong> section after correctly editing the name/path of the configuration file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">configPath</span> <span class="o">&lt;-</span> <span class="s2">&quot;config/config-CNVseq.yml&quot;</span>
</pre></div>
</div>
<p>Load the contents of the configuration file by running the cell,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">config</span> <span class="o">&lt;-</span> <span class="n">read</span><span class="o">.</span><span class="n">config</span><span class="p">(</span><span class="n">configPath</span><span class="p">)</span>
</pre></div>
</div>
<p>Copy number variants (CNVs) are calculated under the <strong>CNV Calculations</strong> section by running the cell,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">cnvCalculate</span><span class="p">(</span><span class="n">config</span><span class="p">)</span>
</pre></div>
</div>
<p>The output tabulated files of all-hits and CNVs can be inspected in the <strong>pbgl-cnvseq/tool/output/PBGL-sorghum-analysis-example/tab-files</strong> directory. These can be used to import data to other softwares of choice for downstream data analysis. A successful run has the following output right under the input cell,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Comparison</span><span class="p">:</span> <span class="n">con</span><span class="o">-</span><span class="mi">2</span><span class="n">_s1</span><span class="o">-</span><span class="n">vs</span><span class="o">-</span><span class="n">D2</span><span class="o">-</span><span class="mi">1</span><span class="n">_S7</span>
<span class="n">Comparing</span> <span class="n">samples</span><span class="p">:</span>
<span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">anibal</span><span class="o">/</span><span class="n">bam_files</span><span class="o">/</span><span class="n">sorghum</span><span class="o">/</span><span class="n">con</span><span class="o">-</span><span class="mi">2</span><span class="n">_S1</span><span class="o">-</span><span class="n">Chromes</span><span class="o">-</span><span class="mi">04</span><span class="o">-</span><span class="mi">05</span><span class="o">-</span><span class="mf">09.</span><span class="n">bam</span>
<span class="n">vs</span><span class="o">.</span>
 <span class="o">/</span><span class="n">home</span><span class="o">/</span><span class="n">anibal</span><span class="o">/</span><span class="n">bam_files</span><span class="o">/</span><span class="n">sorghum</span><span class="o">/</span><span class="n">D2</span><span class="o">-</span><span class="mi">1</span><span class="n">_S7</span><span class="o">-</span><span class="n">Chromes</span><span class="o">-</span><span class="mi">04</span><span class="o">-</span><span class="mi">05</span><span class="o">-</span><span class="mf">09.</span><span class="n">bam</span>

<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;chromosome:  Chr04&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;chromosome:  Chr05&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;chromosome:  Chr09&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  1  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  2  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  3  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  4  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  5  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  6  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  7  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  8  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  9  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  10  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  11  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  12  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  13  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  14  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  15  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  16  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  17  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  18  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  19  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  20  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  21  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  22  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  23  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  24  of  25&quot;</span>
<span class="p">[</span><span class="mi">1</span><span class="p">]</span> <span class="s2">&quot;cnv_id:  25  of  25&quot;</span>

<span class="n">CNV</span> <span class="n">percentage</span> <span class="ow">in</span> <span class="n">genome</span><span class="p">:</span> <span class="mf">1.6</span><span class="o">%</span>
<span class="n">CNV</span> <span class="n">nucleotide</span> <span class="n">content</span><span class="p">:</span> <span class="mi">3140000</span>
<span class="n">CNV</span> <span class="n">count</span><span class="p">:</span> <span class="mi">25</span>
<span class="n">Mean</span> <span class="n">size</span><span class="p">:</span> <span class="mi">125600</span>
<span class="n">Median</span> <span class="n">size</span><span class="p">:</span> <span class="mi">85000</span>
<span class="n">Max</span> <span class="n">Size</span><span class="p">:</span> <span class="mi">750000</span>
<span class="n">Min</span> <span class="n">Size</span><span class="p">:</span> <span class="mi">20000</span>

<span class="n">cnv</span> <span class="n">chromosome</span>      <span class="n">start</span>   <span class="n">end</span>     <span class="n">size</span>    <span class="n">log2</span>    <span class="n">p</span><span class="o">.</span><span class="n">value</span>
<span class="n">CNVR_1</span>      <span class="n">Chr05</span>   <span class="mi">66247501</span>        <span class="mi">66267500</span>        <span class="mi">20000</span>   <span class="mf">1.013742</span>        <span class="mf">1.222196e-59</span>
<span class="n">CNVR_2</span>      <span class="n">Chr09</span>   <span class="mi">23497501</span>        <span class="mi">23527500</span>        <span class="mi">30000</span>   <span class="o">-</span><span class="mf">2.288917</span>       <span class="mf">3.556107e-238</span>
<span class="n">CNVR_3</span>      <span class="n">Chr09</span>   <span class="mi">23547501</span>        <span class="mi">23917500</span>        <span class="mi">370000</span>  <span class="o">-</span><span class="mf">2.684822</span>       <span class="mi">0</span>
<span class="n">CNVR_4</span>      <span class="n">Chr09</span>   <span class="mi">23927501</span>        <span class="mi">24132500</span>        <span class="mi">205000</span>  <span class="o">-</span><span class="mf">2.639143</span>       <span class="mi">0</span>
<span class="n">CNVR_5</span>      <span class="n">Chr09</span>   <span class="mi">24142501</span>        <span class="mi">24302500</span>        <span class="mi">160000</span>  <span class="o">-</span><span class="mf">2.359219</span>       <span class="mi">0</span>
<span class="n">CNVR_6</span>      <span class="n">Chr09</span>   <span class="mi">24337501</span>        <span class="mi">24417500</span>        <span class="mi">80000</span>   <span class="o">-</span><span class="mf">1.907744</span>       <span class="mi">0</span>
<span class="n">CNVR_7</span>      <span class="n">Chr09</span>   <span class="mi">24437501</span>        <span class="mi">24467500</span>        <span class="mi">30000</span>   <span class="o">-</span><span class="mf">0.912225</span>       <span class="mf">1.86383e-68</span>
<span class="n">CNVR_8</span>      <span class="n">Chr09</span>   <span class="mi">24512501</span>        <span class="mi">24652500</span>        <span class="mi">140000</span>  <span class="o">-</span><span class="mf">2.799081</span>       <span class="mi">0</span>
<span class="n">CNVR_9</span>      <span class="n">Chr09</span>   <span class="mi">24657501</span>        <span class="mi">24692500</span>        <span class="mi">35000</span>   <span class="o">-</span><span class="mf">2.466217</span>       <span class="mf">9.479478e-297</span>
<span class="n">CNVR_10</span>     <span class="n">Chr09</span>   <span class="mi">24747501</span>        <span class="mi">24887500</span>        <span class="mi">140000</span>  <span class="o">-</span><span class="mf">2.46515</span>        <span class="mi">0</span>
<span class="n">CNVR_11</span>     <span class="n">Chr09</span>   <span class="mi">24917501</span>        <span class="mi">24942500</span>        <span class="mi">25000</span>   <span class="o">-</span><span class="mf">1.980229</span>       <span class="mf">9.759448e-172</span>
<span class="n">CNVR_12</span>     <span class="n">Chr09</span>   <span class="mi">24947501</span>        <span class="mi">25007500</span>        <span class="mi">60000</span>   <span class="o">-</span><span class="mf">2.378318</span>       <span class="mi">0</span>
<span class="n">CNVR_13</span>     <span class="n">Chr09</span>   <span class="mi">25017501</span>        <span class="mi">25102500</span>        <span class="mi">85000</span>   <span class="o">-</span><span class="mf">2.464295</span>       <span class="mi">0</span>
<span class="n">CNVR_14</span>     <span class="n">Chr09</span>   <span class="mi">25107501</span>        <span class="mi">25132500</span>        <span class="mi">25000</span>   <span class="o">-</span><span class="mf">2.254819</span>       <span class="mf">4.908193e-196</span>
<span class="n">CNVR_15</span>     <span class="n">Chr09</span>   <span class="mi">25137501</span>        <span class="mi">25202500</span>        <span class="mi">65000</span>   <span class="o">-</span><span class="mf">2.90878</span>        <span class="mi">0</span>
<span class="n">CNVR_16</span>     <span class="n">Chr09</span>   <span class="mi">25212501</span>        <span class="mi">25422500</span>        <span class="mi">210000</span>  <span class="o">-</span><span class="mf">3.111295</span>       <span class="mi">0</span>
<span class="n">CNVR_17</span>     <span class="n">Chr09</span>   <span class="mi">25427501</span>        <span class="mi">25512500</span>        <span class="mi">85000</span>   <span class="o">-</span><span class="mf">2.883919</span>       <span class="mi">0</span>
<span class="n">CNVR_18</span>     <span class="n">Chr09</span>   <span class="mi">25522501</span>        <span class="mi">25637500</span>        <span class="mi">115000</span>  <span class="o">-</span><span class="mf">2.617425</span>       <span class="mi">0</span>
<span class="n">CNVR_19</span>     <span class="n">Chr09</span>   <span class="mi">25647501</span>        <span class="mi">25722500</span>        <span class="mi">75000</span>   <span class="o">-</span><span class="mf">2.367854</span>       <span class="mi">0</span>
<span class="n">CNVR_20</span>     <span class="n">Chr09</span>   <span class="mi">25727501</span>        <span class="mi">26477500</span>        <span class="mi">750000</span>  <span class="o">-</span><span class="mf">2.388962</span>       <span class="mi">0</span>
<span class="n">CNVR_21</span>     <span class="n">Chr09</span>   <span class="mi">26482501</span>        <span class="mi">26507500</span>        <span class="mi">25000</span>   <span class="o">-</span><span class="mf">1.727876</span>       <span class="mf">9.894624e-147</span>
<span class="n">CNVR_22</span>     <span class="n">Chr09</span>   <span class="mi">26512501</span>        <span class="mi">26552500</span>        <span class="mi">40000</span>   <span class="o">-</span><span class="mf">1.624917</span>       <span class="mf">3.444569e-216</span>
<span class="n">CNVR_23</span>     <span class="n">Chr09</span>   <span class="mi">26562501</span>        <span class="mi">26787500</span>        <span class="mi">225000</span>  <span class="o">-</span><span class="mf">2.290655</span>       <span class="mi">0</span>
<span class="n">CNVR_24</span>     <span class="n">Chr09</span>   <span class="mi">26817501</span>        <span class="mi">26937500</span>        <span class="mi">120000</span>  <span class="o">-</span><span class="mf">1.704837</span>       <span class="mi">0</span>
<span class="n">CNVR_25</span>     <span class="n">Chr09</span>   <span class="mi">26947501</span>        <span class="mi">26972500</span>        <span class="mi">25000</span>   <span class="o">-</span><span class="mf">1.303558</span>       <span class="mf">1.960969e-100</span>
</pre></div>
</div>
<p>The output tabulated files under <strong>pbgl-cnvseq/tool/output/PBGL-sorghum-analysis-example/tab-files</strong> are the following:</p>
<ul class="simple">
<li><p>con-2_s1-vs-D2-1_S7-all-chroms-CNVs.tab</p></li>
<li><p>con-2_s1-vs-D2-1_S7-window-10000-all-hits.tab</p></li>
</ul>
<p>A depiction of the files are shown below,</p>
<div class="figure align-center" id="id1">
<img alt="all-hits-tab" src="_images/all-hits-tab-file.png" />
<p class="caption"><span class="caption-text">Tabulated file containing all hits of all chromosomes</span><a class="headerlink" href="#id1" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-center" id="id2">
<img alt="cnvs-tab" src="_images/cnvs-tab-file.png" />
<p class="caption"><span class="caption-text">Tabulated file containing all CNVs of all chromsomes</span><a class="headerlink" href="#id2" title="Permalink to this image">¶</a></p>
</div>
<p>After the CNV calculations are done, plot the CNVs under the <strong>Plotting</strong> section. Here, the parameters will be set to the following,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">cnvPlot</span><span class="p">(</span><span class="n">confg</span><span class="p">,</span> <span class="n">imgType</span><span class="o">=</span><span class="s2">&quot;png&quot;</span><span class="p">,</span> <span class="n">yMin</span><span class="o">=-</span><span class="mi">5</span><span class="p">,</span> <span class="n">yMax</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
</pre></div>
</div>
<p>The output images under <strong>pbgl-cnvseq/tool/output/PBGL-sorghum-analysis-example/images</strong> are the following:</p>
<ul class="simple">
<li><p>con-2_s1-vs-D2-1_S7-chromosome-all-window-10000.png</p></li>
<li><p>con-2_s1-vs-D2-1_S7-chromosome-Chr04-window-10000.png</p></li>
<li><p>con-2_s1-vs-D2-1_S7-chromosome-Chr05-window-10000.png</p></li>
<li><p>con-2_s1-vs-D2-1_S7-chromosome-Chr09-window-10000.png</p></li>
</ul>
<p>The output plots are the following,</p>
<div class="figure align-center" id="id3">
<img alt="cnv-all-chroms" src="_images/con-2_s1-vs-D2-1_S7-chromosome-all-window-10000.png" />
<p class="caption"><span class="caption-text">CNV plot of all chromosomes</span><a class="headerlink" href="#id3" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-center" id="id4">
<img alt="cnv-chrom-04" src="_images/con-2_s1-vs-D2-1_S7-chromosome-Chr04-window-10000.png" />
<p class="caption"><span class="caption-text">CNV plot of chromosome Chr04</span><a class="headerlink" href="#id4" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-center" id="id5">
<img alt="cnv-chrom-05" src="_images/con-2_s1-vs-D2-1_S7-chromosome-Chr05-window-10000.png" />
<p class="caption"><span class="caption-text">CNV plot of chromosome Chr05</span><a class="headerlink" href="#id5" title="Permalink to this image">¶</a></p>
</div>
<div class="figure align-center" id="id6">
<img alt="cnv-chrom-09" src="_images/con-2_s1-vs-D2-1_S7-chromosome-Chr09-window-10000.png" />
<p class="caption"><span class="caption-text">CNV plot of chromosome Chr09</span><a class="headerlink" href="#id6" title="Permalink to this image">¶</a></p>
</div>
<p>As seen above, each plot contains:</p>
<ul class="simple">
<li><p>a title of the chromosome being analyzed</p></li>
<li><p>a sub-title of the comparison being done</p></li>
<li><p>Log2 Ratio in the y-axis</p></li>
<li><p>chromosome name on the x-axis in plot of all chromosomes</p></li>
<li><p>base-pair (bp) position on the x-axis in plot of one chromosome</p></li>
<li><p>a label under the x-axis specifying the location of the stored image</p></li>
</ul>
<p>The CNV plot of chromosome Chr09 shows an artifact. Moreover, this artifact seems to lie between base-pairs 2e07 and 3e07. To zoom in, the <strong>Plotting a Zoomed-In Region of One Chromosome</strong> section is provided.</p>
<p>Under the <strong>Plotting a Zoomed-In Region of One Chromosome</strong> section, first specify the location of the all-hits tabulated file of chromosome Chr09,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">tabFile</span> <span class="o">&lt;-</span> <span class="s2">&quot;output/PBGL-sorghum-analysis-example/tab-files/con-2_s1-vs-D2-1_S7-window-10000-all-hits.tab&quot;</span>
</pre></div>
</div>
<p>The run the zooming-in function with the following parameters,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">In</span> <span class="p">[</span> <span class="p">]:</span> <span class="n">cnvPlotZoom</span><span class="p">(</span><span class="n">config</span><span class="p">,</span> <span class="n">tabFile</span><span class="p">,</span> <span class="n">chromosome</span><span class="o">=</span><span class="s2">&quot;Chr09&quot;</span><span class="p">,</span> <span class="n">start</span><span class="o">=</span><span class="mf">2.3e7</span> <span class="p">,</span> <span class="n">end</span><span class="o">=</span><span class="mf">2.8e7</span> <span class="p">,</span> <span class="n">yMin</span><span class="o">=-</span><span class="mi">5</span> <span class="p">,</span> <span class="n">yMax</span><span class="o">=</span><span class="mf">2.5</span><span class="p">,</span> <span class="n">imgType</span><span class="o">=</span><span class="s2">&quot;png&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>The output plot, which is also stored in the <strong>pbgl-cnvseq/tool/output/PBGL-sorghum-analysis-example/images</strong>, shows a zoomed-in region of chromosome Chr09 of sorghum. The output path is also specified under the x-axis label.</p>
<div class="figure align-center" id="id7">
<img alt="cnv-chrom-09-zoom" src="_images/con-2_s1-vs-D2-1_S7-window-10000-chromosome-Chr09-zoomed.png" />
<p class="caption"><span class="caption-text">CNV plot of a zoomed-in region of chromosome Chr09</span><a class="headerlink" href="#id7" title="Permalink to this image">¶</a></p>
</div>
<p>Knowing where artifacts like this one is located plays a major role in identifying insertions or deletions in an organism. The user can, in turn, use different tools, like the Integrative Genomics Viewer (IGV), for further analysis.</p>
<p>This culminates the tutorial.</p>
</div>
</div>
<div class="section" id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">¶</a></h2>
<p><strong>BMC Bioinformatics Publication</strong>:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-80">CNV-seq, a new method to detect copy number variation using high-throughput sequencing</a></p></li>
</ul>
<p><strong>GitHub repositories</strong>:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://github.com/hliang/cnv-seq">hliang/cnv-seq</a></p></li>
<li><p><a class="reference external" href="https://github.com/Bioconductor/copy-number-analysis/wiki/CNV-seq">Bioconductor/copy-number-analysis</a></p></li>
<li><p><a class="reference external" href="https://github.com/pbgl/pbgl-cnvseq">pbgl/pbgl-cnvseq</a></p></li>
<li><p><a class="reference external" href="https://github.com/amora197/pbgl-cnvseq">amora197/pbgl-cnvseq</a></p></li>
</ul>
</div>
</div>


           </div>
           
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, H. Liang, Bioconductor, Anibal E. Morales.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>